---
title: "KelsoTTRAnalysis"
output: html_document
date: '2023-07-30'
---

In this document I'll examine how time-to-right (TTR) varies by temperature, time, and genotype for Julia's data.

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../../output/repeat-heat/03-TTR-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, include=FALSE}
#install.packages("tidyverse")
#install.packages("plotrix")
#install.packages("RColorBrewer")
#install.packages(lme4)
#install.packages("ggpubr")
#install.packages("ggrain")
#install.packages("cowplot")
require(tidyverse)
require(plotrix)
require(RColorBrewer)
require(lme4)
require(ggpubr)
require(ggrain)
require(cowplot)
```

```{r}
sessionInfo()
```

# Import data

```{r}
rawTTR <- read.csv("../../../data/repeat-heat/time-to-right-genotype-julia.csv", header = TRUE) #Import raw data
head(rawTTR) #Confirm import. Trial data is in seconds.
```

# Format data

```{r}
modTTR <- rawTTR %>%
  dplyr::select(., -c(notes, probe.number)) %>%
  mutate(., day = case_when(date == "7/31/23" ~ 1,
                            date == "8/1/23" ~ 2,
                            date == "8/2/23" ~ 2,
                            date == "8/4/23" ~ 5)) %>%
  mutate(., treatment = case_when (tank == "10" | tank == "12" ~ "No Pulse",
                                   tank == "11" | tank == "13" ~ "Pulse")) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  filter(., is.na(TTRSE) == FALSE) %>%
  group_by(., day, treatment) %>%
  mutate(., TTRavgFull = mean(TTRavg)) %>%
  mutate(., TTRSEFull = std.error(TTRavg)) %>%
  mutate(., TTRavgFullLow = TTRavgFull - TTRSEFull) %>%
  mutate(., TTRavgFullHigh = TTRavgFull + TTRSEFull) %>%
  ungroup(.) #Remove sampling ID and notes columns. Add new column with day information, combining 8/1 and 8/2 into the same day4. Create a a new column that with an index for width/length. Create new column as a binary for whether or not crabs are missing legs. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations, and remove rows where TTRavg | TTRSE = NA. Calculate average and SE for all samples in a treatment for each day. Add/subtract TTRSEFull to/from TTRavgFull to get bounds. Ungroup.a.
head(modTTR) #Confirm formatting
```

# Population assessment

Before proceeding, I want to understand the distribution of genotypes in the population. I also need to get the average weight and carapace width of the crabs.

```{r}
modTTR[1:24]$carapace.width %>% mean(., na.rm = TRUE) #46.83333
modTTR[1:24]$carapace.width %>% sd(., na.rm = TRUE) #5.505439

modTTR[1:24]$weight %>% mean(., na.rm = TRUE) #24.17931
modTTR[1:24]$weight %>% sd(., na.rm = TRUE) #8.280845
```

```{r}
modTTR %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.4444
ggsave("figures/genotype-pie-chart.jpg", height = 8.5, width = 11)
```

# Hardy-Weinberg equilibrium

Since I have the genotype data, I want to see if my population is in Hardy-Weinbergy equilibrium. I will need to do this for the full MA population, as well as for the individual New Bedford and Eel Pond populations. I used [this calculator](https://www.cog-genomics.org/software/stats) to obtain p-values.

```{r}
modTTR %>%
  group_by(., genotype) %>%
  summarize(., count = n())
```

P-value: 0.32976593218497

This population is in Hardy-Weinberg equilibrium!

# Assess differences in average TTR

## Mixed effects model

In addition to crab ID, I will need to include tank as a random effect since there were significant differences in tank temperatures during the second pulse.

```{r}
hist(modTTR$TTRavg)
hist(x = log(modTTR$TTRavg)) #Log transformation helps normalize data
```

```{r}
TTRmodel <- lmer(log(TTRavg) ~ as.factor(treatment) + day + as.factor(treatment):day + 
                   as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                   (1|crab.ID) + (1|tank), 
                 data = modTTR, REML = FALSE) #Run model with treatment, timepoint, their interaction, sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
summary(TTRmodel) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
var.lmer <- c(0.02054, 0, 0.22874) #Save variances of random effects as a new vector (crab ID, tank, residuals)
percentvar.lmer <- (100*var.lmer)/sum(var.lmer) #Calculate percent variances
percentvar.lmer[1] #Crab ID accounts for 8.23973% of variance
percentvar.lmer[2] #Tank accounts for 0% of the variance!
```

Now that I've run the full model, I want to test the importance of treatment, day, and their interaction as predictors. I'll include all other factors in the null model to account for any variation they may contribute.

```{r}
TTRnullTreatment <- lmer(log(TTRavg) ~ day + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run null model without treatment to identify significance of predictor
anova(TTRnullTreatment, TTRmodel) #Treatment is a significant predictor
```

```{r}
TTRnullDay <- lmer(log(TTRavg) ~ as.factor(treatment) + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run null model without day to identify significance of predictor
anova(TTRnullDay, TTRmodel) #Day is a significant predictor
```

```{r}
TTRnullInteraction <- lmer(log(TTRavg) ~ as.factor(treatment) + day + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run model without interaction to identify significance of predictor. The full model constructed does not include an interaction
anova(TTRnullInteraction, TTRmodel) #Interaction is a significant predictor
```

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullTreatment, TTRmodel)),
                            broom::tidy(anova(TTRnullDay, TTRmodel)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodel))) #Create table for model comparison output
TTRModelComparisons
```

```{r}
TTRModelComparisons %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) #Perform bonferroni adjustment for multiple comparisons
```

### Calculate average TTR

```{r}
modTTR %>%
  group_by(., treatment, day) %>%
  summarize(., mean = mean(TTRavg), sd = sd(TTRavg)) %>%
  write.csv("average-TTR-calculations.csv", quote = FALSE, row.names = FALSE) #Take TTR data and group by treatment and day. Calculate mean and standard deviation. Save as a csv file.
```

## Check assumptions

```{r}
hist(residuals(TTRmodel)) #Residuals normally distributed

plot(fitted(TTRmodel), residuals(TTRmodel)) #Borderline heteroskedastic but probably okay? Dispersion across x-axis is likely related to discrete variables
abline(h = 0, lty = 2, col = "grey")
```

## Effect of genotype

Now that I know treatment, time, and their interaction significantly impact TTR, I want to add in genotype and sourcing information.

```{r}
TTRfullGenotype <- lmer(log(TTRavg) ~ as.factor(treatment) + day + as.factor(treatment):day + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(genotype) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run model with genotype to identify4 significance of predictor
anova(TTRmodel, TTRfullGenotype) #No significant impact of genotype
```

```{r}
TTRpresenceC <- lmer(log(TTRavg) ~ as.factor(treatment) + day + as.factor(treatment):day + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceC) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodel, TTRpresenceC) #No significant impact of C allele
```

```{r}
TTRpresenceT <- lmer(log(TTRavg) ~ as.factor(treatment) + day + as.factor(treatment):day + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceT) + (1|crab.ID) + (1|tank), data = modTTR, REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodel, TTRpresenceT) #Marginal impact of T allele
```

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullTreatment, TTRmodel)),
                            broom::tidy(anova(TTRnullDay, TTRmodel)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodel)),
                            broom::tidy(anova(TTRmodel, TTRfullGenotype)),
                            broom::tidy(anova(TTRmodel, TTRpresenceC)),
                            broom::tidy(anova(TTRmodel, TTRpresenceT))) #Create table for model comparison output
TTRModelComparisons
```

```{r}
TTRModelComparisons <- TTRModelComparisons %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Perform p-value corrections
TTRModelComparisons
```

```{r}
write.csv(TTRModelComparisons, "ttr-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save model output
```

The marginally significant impact of the T allele on average TTR is no longer significant after p-value correction. However, I'll still include this information in a plot since there is a trend. 

### Calculate average TTR

```{r}
modTTR %>%
  group_by(., treatment, day, presenceT) %>%
  summarize(., mean = mean(TTRavg), sd = sd(TTRavg)) %>%
  write.csv("average-TTR-calculations-genotype.csv", quote = FALSE, row.names = FALSE) #Take TTR data and group by treatment and day. Calculate mean and standard deviation. Save as a csv file.
```

```{r}
modTTR %>%
  group_by(., treatment, day, presenceT) %>%
  summarize(., mean = mean(TTRavg), sd = sd(TTRavg)) %>%
  filter(., day == 5)
```


My next step is to understand how the difference in TTR is impacted by treatment, time, and genotype.

# Assess impacts on differences in TTR

To understand how various factors impact the difference in TTR, I need to reformat my dataframe.

```{r}
modTTRWide <- modTTR %>%
  select(-c(date, trial.1:trial.3, integument.color:carapace.length, integument.cont, TTRSE:TTRavgFullHigh)) %>%
  pivot_wider(., names_from = "day", names_prefix = "day", values_from = "TTRavg") %>%
  select(., -c(day2:day3)) %>% 
  mutate(., TTRdiff = day5 - day1) %>%
  left_join(x = ., 
            y = (modTTR %>%
                   filter(., day == 5) %>%
                   select(., crab.ID, integument.cont, weight, carapace.width, missing.swimmer)), 
            by = "crab.ID") #Take data and remove extraneous columns. Pivot dataframe and remove intermediate timepoints to have day 1 (before the experiment) and day 5 (after every crab experienced the 24 hr pulse). Calculate the difference in TTR between timepoints. Add demographic information from the final sampling point back to the dataframe.
modTTRWide #Confirm changes
```

```{r}
hist(modTTRWide$TTRdiff)
hist(modTTRWide$TTRdiff %>% log(.))
```

## Linear model

Since I don't have multiple timepoints for each crab, I don't need to include individual as a random effect. Therefore, I can run a general linear model instead of a mixed effects model.

```{r}
TTRDiffModel <- lm(log(TTRdiff) ~ as.factor(treatment) +
                     as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer), 
                   data = modTTRWide) #Run model with treatment, timepoint, their interaction, sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
summary(TTRDiffModel) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

Now that I've run the full model, I want to test the importance of treatment, day, and their interaction as predictors. I'll include all other factors in the null model to account for any variation they may contribute.

```{r}
step(TTRDiffModel, direction = "backward")
```


```{r}
TTRDiffModel2 <- lm(formula = log(TTRdiff) ~ as.factor(treatment) + integument.cont + 
                      carapace.width + weight + as.factor(missing.swimmer), data = modTTRWide) #Model identified by step
summary(TTRDiffModel2) #Summary of model information from step
```

## Effect of genotype

Now that I have a significant base model, I'm going to add genotype information to it.

```{r}
TTRDiffModel3 <- lm(formula = log(TTRdiff) ~ as.factor(treatment) + integument.cont + 
                      carapace.width + weight + as.factor(missing.swimmer) +
                      genotype, data = modTTRWide) #Model identified by step + genotype
summary(TTRDiffModel3)
```

```{r}
TTRDiffModel4 <- lm(formula = log(TTRdiff) ~ as.factor(treatment) + integument.cont + 
                      carapace.width + weight + as.factor(missing.swimmer) +
                      presenceC, data = modTTRWide) #Model identified by step + presenceC
summary(TTRDiffModel4)
```

```{r}
TTRDiffModel5 <- lm(formula = log(TTRdiff) ~ as.factor(treatment) + integument.cont + 
                      carapace.width + weight + as.factor(missing.swimmer) +
                      presenceT, data = modTTRWide) #Model identified by step + presenceC
summary(TTRDiffModel5)
```

No significant effect of any genotype variable on differences in TTR! Also interestingly no significant effect of pre-conditioning on differences.

# Plots

```{bash}
mkdir figures
```

```{r}
plotColors <- c(brewer.pal(9, "Reds")[7],
                brewer.pal(9, "Greys")[7]) #Create color scheme
```

## Average TTR plot

```{r}
modTTR %>%
  filter(., day == 1 | day == 5) %>%
  ggplot(mapping = aes(x = as.factor(day), y = TTRavg, fill = treatment)) +
  geom_rain(alpha = 0.5, id.long.var = "crab.ID") +
  facet_wrap(~treatment, scale = "free_y", nrow = 2) +
  scale_x_discrete(name = "",
                   labels = c("Before", "After")) +
  scale_y_continuous(name = "Average TTR (s)") +
  scale_fill_manual(name = "Treatment",
                    values = c('#9699eb', '#eb8d88')) +
  scale_shape_manual(name = "Genotype",
                     labels = c("CC", "CT or TT"),
                     values = c(17, 19)) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/average-TTR-raincloud-lines.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(., day == 1 | day == 5) %>%
  ggplot(mapping = aes(x = as.factor(day), y = TTRavg, fill = treatment)) +
  geom_rain(alpha = 0.5) +
  facet_wrap(~treatment, scale = "free_y", nrow = 2) +
  scale_x_discrete(name = "",
                   labels = c("Before", "After")) +
  scale_y_continuous(name = "Average TTR (s)") +
  scale_fill_manual(name = "Treatment",
                    values = c('#9699eb', '#eb8d88')) +
  scale_shape_manual(name = "Genotype",
                     labels = c("CC", "CT or TT"),
                     values = c(17, 19)) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/average-TTR-raincloud.pdf", height = 8.5, width = 11)
```

```{r}
mainPlot <- modTTR %>%
  dplyr::select(., c(day, treatment, TTRavg, presenceT)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(day), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  annotate("rect", 
           xmin = 2, xmax = 2.38, 
           ymin = 0, ymax = 15, 
           alpha = 0.2, fill = plotColors[1]) +
  annotate("rect", 
           xmin = 2.62, xmax = 3.38, 
           ymin = 0, ymax = 15, 
           alpha = 0.2, fill = plotColors[1]) +
  scale_x_discrete(name = "Day",
                   labels = c("0", "1", "4")) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(0:5, 10, 15)) +
  scale_color_manual(name = "",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_shape_manual(values = c(15, 19),
                     name = "",
                     labels = c("Control", "Treatment")) +
  theme_classic(base_size = 15) #Select unique day, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.

insetPlot <-  modTTR %>%
  dplyr::select(., c(day, treatment, TTRavg, presenceT)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(day), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  ylab("") +
  scale_x_discrete(name = "Day",
                   labels = c("0", "1", "4")) +
  scale_color_manual(name = "",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_y_continuous(limits = c(0,4)) +
  scale_shape_manual(values = c(15, 19),
                     name = "",
                     labels = c("Control", "Treatment")) +
  theme_classic() + theme(legend.position = "none",
                          axis.line.x.bottom = element_line(linewidth = 0),
                          axis.line.y.left = element_line(linewidth = 0),
                          panel.border = element_rect(colour = "black", fill = NA, size = 1)) #Create an inset plot for 0 < TTR < 4

plotWithInset <- ggdraw() +
  draw_plot(mainPlot) +
  draw_plot(insetPlot, x = 0.08, y = 0.71, width = 0.52, height = 0.25) #Add inset to the plot

plotWithInset #Plot the full plot with inset

ggsave("figures/time-to-right-avg-boxplot-wInset.pdf", height = 8.5, width = 11) #Save!
```

## Genotype-specific plots

```{r}
presenceT_labels <- c("N" = "CC",
                      "Y" = "CT or TT") #Modify facet labels for precense of the T allele
```

```{r}
modTTR %>%
  ggplot(mapping = aes(x = as.factor(day), y = TTRavg, color = treatment), shape = treatment) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(shape = treatment), position = position_dodge(width = 0.1)) +
  facet_grid(treatment~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "Day",
                   labels = c(0, 1, 4)) +
  scale_y_continuous(name = "Average Time-to-Right (s)") +
  scale_color_manual(name = "",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_shape_manual(name = "",
                     values = c(15, 19),
                     labels = c("Control", "Treatment")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"),
                                        strip.text.y = element_blank())
ggsave("figures/average-TTR-boxplot-treatT.pdf", height = 8.5, width = 11)
```

## Difference in TTR plot

```{r}
modTTRWide %>%
  ggplot(mapping = aes(x = treatment, y = TTRdiff, color = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(shape = presenceT), position = position_dodge(width = 0.1)) +
  facet_grid(~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "",
                   labels = c("No Pulse", "Pulse")) +
  scale_y_continuous(name = "Difference in TTR (s)") +
  scale_color_manual(guide = "none",
                     values = c('#9699eb', '#eb8d88')) +
  scale_shape_manual(guide = "none",
                     values = c(17, 19)) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/difference-TTR-boxplot-treatT.pdf", height = 8.5, width = 11)
```

```{r}
modTTRWide %>%
  ggplot(mapping = aes(x = treatment, y = TTRdiff, color = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(size = carapace.width, alpha = 0.5), position = position_dodge(width = 0.1)) +
  facet_grid(~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "",
                   labels = c("No Pulse", "Pulse")) +
  scale_y_continuous(name = "Difference in TTR (s)") +
  scale_color_manual(guide = "none",
                     values = c('#9699eb', '#eb8d88')) +
  scale_size(name = "Carapace Width", range = c(1,8)) +
  scale_alpha(guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/difference-TTR-boxplot-treatTwidth.pdf", height = 8.5, width = 11)
```

```{r}
modTTRWide %>%
  ggplot(mapping = aes(x = treatment, y = TTRdiff, color = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(size = weight, alpha = 0.5), position = position_dodge(width = 0.1)) +
  facet_grid(~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "",
                   labels = c("No Pulse", "Pulse")) +
  scale_y_continuous(name = "Difference in TTR (s)") +
  scale_color_manual(guide = "none",
                     values = c('#9699eb', '#eb8d88')) +
  scale_size(name = "Weight", range = c(1,10)) +
  scale_alpha(guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/difference-TTR-boxplot-treatTweight.pdf", height = 8.5, width = 11)
```

```{r}
modTTRWide %>%
  ggplot(mapping = aes(x = treatment, y = TTRdiff, color = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(shape = missing.swimmer), alpha = 0.8, size = 3, position = position_dodge(width = 0.1)) +
  facet_grid(~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "",
                   labels = c("No Pulse", "Pulse")) +
  scale_y_continuous(name = "Difference in TTR (s)") +
  scale_color_manual(guide = "none",
                     values = c('#9699eb', '#eb8d88')) +
  scale_shape(name = "Missing Swimmer?",
              labels = c("No", "Yes")) +
  scale_alpha(guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"))
ggsave("figures/difference-TTR-boxplot-treatTswimmer.pdf", height = 8.5, width = 11)
```

## Multipanel plot

### Modify plots for multipanel

```{r}
mainPlotMultipanel <- modTTR %>%
  dplyr::select(., c(day, treatment, TTRavg, presenceT)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(day), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  annotate("rect", 
           xmin = 2, xmax = 2.38, 
           ymin = 0, ymax = 15, 
           alpha = 0.2, fill = plotColors[1]) +
  annotate("rect", 
           xmin = 2.62, xmax = 3.38, 
           ymin = 0, ymax = 15, 
           alpha = 0.2, fill = plotColors[1]) +
  scale_x_discrete(name = "",
                   labels = c("0", "1", "4")) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(0:5, 10, 15)) +
  scale_color_manual(guide = "none",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_shape_manual(guide = "none",
                     values = c(15, 19),
                     labels = c("Control", "Treatment")) +
  ggtitle("A. Average Righting Response") +
  theme_classic(base_size = 15) + theme(legend.position = "none") #Select unique day, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.

insetPlotMultipanel <-  modTTR %>%
  dplyr::select(., c(day, treatment, TTRavg, presenceT)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(day), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  ylab("") +
  scale_x_discrete(name = "Day",
                   labels = c("0", "1", "4")) +
  scale_color_manual(guide = "none",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_y_continuous(limits = c(0,4)) +
  scale_shape_manual(guide = "none",
                     values = c(15, 19),
                     labels = c("Control", "Treatment")) +
  theme_cowplot() + theme(legend.position = "none",
                          axis.line.x.bottom = element_line(linewidth = 0),
                          axis.line.y.left = element_line(linewidth = 0),
                          panel.border = element_rect(colour = "black", fill = NA, size = 1),
                          plot.margin = unit(c(0,0,0,0), "pt")) #Create an inset plot for 0 < TTR < 4

plotWithInsetMultipanel <- ggdraw() +
  draw_plot(mainPlotMultipanel) +
  draw_plot(insetPlotMultipanel, x = 0.075, y = 0.485, width = 0.44, height = 0.4) #Add inset to the plot

plotWithInsetMultipanel #Plot the full plot with inset
```

```{r}
genotypeMultipanel <- modTTR %>%
  ggplot(mapping = aes(x = as.factor(day), y = TTRavg, color = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(aes(shape = treatment), position = position_dodge(width = 0.1)) +
  facet_grid(treatment~presenceT, scale = "free_y", labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_discrete(name = "Day",
                   labels = c(0, 1, 4)) +
  scale_y_continuous(name = "Average Time-to-Right (s)") +
  scale_color_manual(name = "",
                     values = c(plotColors[2], plotColors[1]),
                     labels = c("Control", "Treatment")) +
  scale_shape_manual(name = "",
                     values = c(15, 19),
                     labels = c("Control", "Treatment")) +
  ggtitle("B. Righting Response by Genotype") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"),
                                        strip.text.y = element_blank(),
                                        legend.position = "bottom")
```


```{r}
plot_grid(plotWithInsetMultipanel, genotypeMultipanel, ncol = 1) #Use plot_grid from cowplot, not patchwork, to create a multipanel due to the inset being a cowplot object. Align vertically by using 1 column
ggsave("figures/multipanel-TTR-plot.pdf", height = 11, width = 8.5) #Save multipanel plot
```


