---
title: "03-TTR-analysis-MA"
author: "Yaamini Venkataraman"
date: '2024-07-12'
output: html_document
---

In this document I'll examine how time-to-right (TTR) varies by temperature and time for MA crabs.

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../../output/MA/03-TTR-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, include=FALSE}
#install.packages("tidyverse")
#install.packages("plotrix")
#install.packages("RColorBrewer")
#install.packages("lme4")
# install.packages("emmeans")
# install.packages("multcompView")
# install.packages("multcomp")
#install.packages("ggpubr")
#install.packages("patchwork")
#install.packages("respirometry")
require(tidyverse)
require(plotrix)
require(RColorBrewer)
require(lme4)
require(emmeans)
require(multcompView)
require(multcomp)
require(ggpubr)
require(patchwork)
require(respirometry)
```

```{r}
sessionInfo()
```

# Import data

```{r}
rawTTR <- read.csv("../../../data/MA/time-to-right-MA-clean.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTR) #Confirm import. Trial data is in seconds.
```

```{r}
crabMetadata <- read.csv("../../../data/MA/crab-metadata-MA.csv", header = TRUE) %>%
  dplyr::select(., c("number", "tank", "carapace.width", "carapace.length")) %>%
  dplyr::rename(crab.ID = "number") #Import metadata. Select ID, treatment, width and length columns. Rename "ID" as "crab.ID"
head(crabMetadata) #Confirm changes
```

```{r}
genotypeData <- read.csv("../../../data/MA/2024-crab-genotype-MA.csv", header = TRUE, na.strings = "") %>%
  dplyr::select(., -c(extraction.date:gel.date, notes:QC.gen)) %>%
  mutate(., crab.ID = gsub(pattern = "MA-", replacement = "", x = ID)) %>%
  mutate(., crab.ID = as.numeric(crab.ID)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeData) #Confirm genotype dataframe creation
```

# Format data

```{r}
modTTR <- rawTTR %>%
  dplyr::select(., -notes) %>%
  left_join(., crabMetadata, by = c("crab.ID", "tank", "carapace.width")) %>%
  filter(., is.na(trial.1) == FALSE) %>%
  mutate(., integument.cont = case_when(integument.color == "B/G" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "Y/G" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "Y/O" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "R/O" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., treatment = case_when(tank == "T01" ~ "0C",
                                  tank == "T02" ~ "0C",
                                  tank == "T03" ~ "0C",
                                  tank == "T05" ~ "5C",
                                  tank == "T06" ~ "5C",
                                  tank == "T07" ~ "5C")) %>%
  mutate(., missing.swimmer = case_when(missing.swimmer == "Y" ~ missing.swimmer,
                                        missing.swimmer == "N" ~ missing.swimmer,
                                        missing.swimmer == "Y " ~ "Y")) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeData %>% dplyr::select(source:presenceT)), by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTR) #Confirm formatting
```

# Data exploration

```{bash}
mkdir figures
```

```{r}
plotColors <- c(brewer.pal(9, "Blues")[9],
                brewer.pal(9, "Blues")[7]) #Create color scheme
```

## Treatment

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = treatment)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "Timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]), 
                     name = "Temperature (ºC)",
                     breaks = c("0C", "5C"),
                     labels = c("0", "5")) +
  theme_classic(base_size = 15) #Plot average TTR and add lm fits without SE bars. Modify x-axis to only show experimental timepoints where TTR was measured. Assign colors to each treatment. Increase base font size.
ggsave("figures/time-to-right-raw.pdf", height = 8.5, width = 11)
```

I'm now going to make plots for various other factors to see if any of these need to be taken into consideration!

## Sex

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = sex)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-sex.pdf", height = 8.5, width = 11)
```

## Tank

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = tank)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-tank.pdf", height = 8.5, width = 11)
```

## Integument color

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = integument.color)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  scale_color_manual(values = c("#66C2A5","#A6D96A","#D9EF8B","#FFFF99","#FDAE61","#F46D43","#D53E4F"),
                     name = "Integument",
                     breaks = c("B/G", "G", "Y/G", "Y", "Y/O", "O", "R/O"),
                     labels = c("B/G", "G", "Y/G", "Y", "Y/O", "O", "R/O")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-integument.pdf", height = 8.5, width = 11)
```

## Missing swimmer

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = as.factor(missing.swimmer))) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-missswim.pdf", height = 8.5, width = 11)
```

## Weight

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = weight)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-weight.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight

#Weight = 23.83123 g ± 14.52899 g
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  group_by(., source) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight by subpopulation

#EP = 20.62880 g ± 11.62681 g
#NB = 25.26089 g ± 15.53417 g
```

## Carapace width

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = carapace.width)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 65, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-carapaceWidth.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE), 
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width

#CW: 46.23171 mm ± 8.189573 mm
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  group_by(., source) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE), 
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width by subpopulation

#EP: 44.19231	mm ± 7.647322	mm
#NB: 47.17857	mm ± 8.325442	mm
```

## Genotype

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = genotype)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 95, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-genotype.pdf", height = 8.5, width = 11)
```

```{r}
genotypeData %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart.jpg", height = 8.5, width = 11)
```

```{r}
genByPop <- genotypeData %>%
  group_by(., source, genotype) %>%
  summarize(., count = n()) #Take genotype data and group by population and genotype. 
genByPop #Confirm formatting
```

```{r}
genByPop %>%
  filter(., source == "EP") %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart-EP.jpg", height = 8.5, width = 11)
```

```{r}
genByPop %>%
  filter(., source == "NB") %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart-NB.jpg", height = 8.5, width = 11)
```

### Hardy-Weinberg equilibrium

Since I have the genotype data, I want to see if my population is in Hardy-Weinbergy equilibrium. I will need to do this for the full MA population, as well as for the individual New Bedford and Eel Pond populations. I used [this calculator](https://www.cog-genomics.org/software/stats) to obtain p-values.

Full population p-value: 0.0000071191801649826
New Bedford p-value: 0.02291149613108
Eel Pond p-value: 6.6098836727653e-7

No subset of this MA population is in Hardy-Weinberg equilibrium!

# Assess differences in TTR

I will use a linear mixed effects model with individual as a random effect to analyze the data, following [this tutorial](https://jontalle.web.engr.illinois.edu/MISC/lme4/bw_LME_tutorial.pdf).

```{r}
hist(modTTR$TTRavg)
hist(x = log(modTTR$TTRavg)) #Log transformation helps normalize data
```

## Assess importance of demographic variables

```{r}
TTRmodelDV <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
```

```{r}
TTRnullSex <- lmer(log(TTRavg) ~ integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRnullSex, TTRmodelDV)
```

```{r}
TTRnullIntegument <- lmer(log(TTRavg) ~ as.factor(sex) + carapace.width + weight + as.factor(missing.swimmer) + 
                            (1|crab.ID), 
                          data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRnullIntegument, TTRmodelDV) #Integument color is not significant
```

```{r}
TTRnullCW <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + weight + as.factor(missing.swimmer) + 
                    (1|crab.ID), 
                  data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRnullCW, TTRmodelDV) #CW is not significant
```

```{r}
TTRnullWeight <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRnullWeight, TTRmodelDV) #Weight is not significant
```

```{r}
TTRnullSwimmer <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight +
                         (1|crab.ID), 
                       data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRnullSwimmer, TTRmodelDV) #Missing swimmer is not significant
```

No demographic variables significantly impact TTR.

## Assess importance of explanatory variables

```{r}
TTRmodelEV <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                     (1|crab.ID), 
                   data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with treatment, timepoint, and their interaction
```

```{r}
TTRnullTreatment <- lmer(log(TTRavg) ~ timepoint + 
                           (1|crab.ID), 
                         data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run null model without treatment to identify significance of predictor
anova(TTRnullTreatment, TTRmodelEV) #Treatment is a significant predictor
```

```{r}
TTRnullTimepoint <- lmer(log(TTRavg) ~ treatment +
                           (1|crab.ID),
                         data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullTimepoint, TTRmodelEV) #Timepoint is a significant predictor
```

```{r}
TTRnullInteraction <- lmer(log(TTRavg) ~ treatment + timepoint + 
                             (1|crab.ID),
                           data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without interaction to identify significance of predictor. The full model constructed does not include an interaction
anova(TTRnullInteraction, TTRmodelEV) #Interaction is a significant predictor
```

## Assess importance of genotype

Now that I know treatment, time, and their interaction significantly impact TTR, I want to add in genotype and sourcing information.

```{r}
TTRfullGenotype <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(genotype) + 
                          (1|crab.ID),
                        data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelEV, TTRfullGenotype) #No significant impact of genotype
```

```{r}
TTRpresenceC <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                       as.factor(presenceC) + 
                       (1|crab.ID),
                     data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelEV, TTRpresenceC) #No significant impact of C allele
```

```{r}
TTRpresenceT <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                       as.factor(presenceT) + 
                       (1|crab.ID), 
                     data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelEV, TTRpresenceT) #No significant impact of T allele
```

```{r}
TTRsource <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                    as.factor(source) + 
                    (1|crab.ID), 
                  data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelEV, TTRsource) #No significant impact of source on TTR
```

## Final model

```{r}
summary(TTRmodelEV)
```


```{r}
var.lmer <- c(0.05169, 0.41876) #Save variances of random effects as a new vector (crab ID, residual)
percentvar.lmer <- (100*var.lmer)/sum(var.lmer) #Calculate percent variances
percentvar.lmer[1] #Crab ID accounts for 10.98735% of variance
```

## Save model output

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullSex, TTRmodelDV)),
                            broom::tidy(anova(TTRnullIntegument, TTRmodelDV)),
                            broom::tidy(anova(TTRnullCW, TTRmodelDV)),
                            broom::tidy(anova(TTRnullWeight, TTRmodelDV)),
                            broom::tidy(anova(TTRnullSwimmer, TTRmodelDV)),
                            broom::tidy(anova(TTRnullTreatment, TTRmodelEV)),
                            broom::tidy(anova(TTRnullTimepoint, TTRmodelEV)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodelEV)),
                            broom::tidy(anova(TTRmodelEV, TTRfullGenotype)),
                            broom::tidy(anova(TTRmodelEV, TTRpresenceC)),
                            broom::tidy(anova(TTRmodelEV, TTRpresenceT)),
                            broom::tidy(anova(TTRmodelEV, TTRsource))) %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Create table for model comparison output. Adjust p-values for multiple comparisons
TTRModelComparisons
```

```{r}
write.csv(TTRModelComparisons, "ttr-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save model output
```

## Posthoc tests for significant predictors

There were significant impacts of treatment, timepoint, and their interaction on average righting response! I will investigate these effects using post-hoc tests with `emmeans`.

### Treatment

The first thing I will do is explore pairwise differences between treatment. Since time was also significant, I'll need to look at treatment differences at ech timepoint.

```{r}
TTRmodel_treatment <- emmeans(TTRmodelEV, pairwise ~ treatment | timepoint, at = list(timepoint = c(0, 1, 2, 3, 4, 5)), adjust = "tukey", data = (modTTR %>% filter(., is.na(TTRavg) == FALSE))) #Use emmeans to obtain pairwise treatment differences at the timepoints listed. 
TTRmodel_treatment$contrasts %>% broom::tidy(.) #Extract the contrast information to look at pairwise test output. Clean output using broom.
TTRmodel_treatment$contrasts %>% 
  broom::tidy(.) %>%
  write.csv("tukey-test-treatment.csv", quote = FALSE, row.names = FALSE) #Save output as a csv.
```

Interactions can also be investigated by looking at differences in linear model slopes. I will estimate simple slopes to understand how average TTR changes over time for each treatment.

```{r}
TTRmodel_treatment_slope <- emtrends(TTRmodelEV, pairwise ~ treatment, var = "timepoint", data = (modTTR %>% filter(., is.na(TTRavg) == FALSE))) #Obtain simple slopes for the interaction between each temperature and time. Test if the effect of time significantly differs between the two treatments
TTRmodel_treatment_slope$emtrends #Different slopes for each temperature (makes sense because there was an interaction)
TTRmodel_treatment_slope$contrasts #Significance test for the different slopes.
```

### Timepoint

I want to understand how TTR varied by timepoint within different treatments. There is no way to do this with my original model since timepoint was (rightfully) coded as a continuous variable. In order to get pairwise estimates, I will need to refactor time as a categorical variable, rerun the model, and then use `emmeans` on this revised model.

```{r}
modTTR$timepoint_factor <- as.factor(modTTR$timepoint) #Code timepoint as a factor
TTRmodel_factor <- lmer(log(TTRavg) ~ treatment + timepoint_factor + treatment:timepoint_factor + 
                          (1|crab.ID), 
                        data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run the original model, but use timepoint as a factor
TTRmodel_time <- emmeans(TTRmodel_factor, pairwise ~ timepoint_factor | treatment, adjust = "tukey", data = (modTTR %>% filter(., is.na(TTRavg) == FALSE))) #Rerun emmeans using timepoint as a factor
TTRmodel_time$contrasts %>% broom::tidy(.) #Extract contrasts to look at pairwise output. Clean output using broom.
TTRmodel_time$contrasts %>%
  broom::tidy(.) %>%
  write.csv("tukey-test-timepoint.csv", quote = FALSE, row.names = FALSE) #Save output as a csv.
```

## Calculate average values

```{r}
modTTR %>%
  group_by(., treatment, timepoint) %>%
  summarize(mean = mean(TTRavg, na.rm = TRUE), sd = sd(TTRavg, na.rm = TRUE)) %>%
  write.csv("average-TTR-calculations.csv", quote = FALSE, row.names = FALSE)
```

## Sensitivity test

We sampled crabs throughout the experiment, so some crabs were removed from the study. I am going to repeat the modeling efforts using only the crabs that were alive throughout the experiment.

```{r}
modTTRSubset <- modTTR %>%
  dplyr::select(., timepoint, crab.ID, TTRavg, treatment) %>%
  group_by(., crab.ID) %>%
  pivot_wider(., names_from = timepoint, values_from = TTRavg, names_prefix = "timepoint") %>%
  mutate_all(~ifelse(is.nan(.), 91, .)) %>%
  filter(., is.na(timepoint0) == FALSE) %>%
  filter(., is.na(timepoint1) == FALSE) %>%
  filter(., is.na(timepoint2) == FALSE) %>%
  filter(., is.na(timepoint3) == FALSE) %>%
  filter(., is.na(timepoint4) == FALSE) %>%
  filter(., is.na(timepoint5) == FALSE) %>%
  pivot_longer(., cols = timepoint0:timepoint5, names_to = "timepoint", values_to = "TTRavg") %>%
  mutate(., timepoint = gsub(x = timepoint, pattern = "timepoint", replacement = "") %>% as.numeric) %>%
  left_join(., y = modTTR, by = c("crab.ID", "timepoint", "treatment")) %>%
  dplyr::select(., -c(date:tank, TTRavg.y:source)) %>%
  dplyr::rename(., TTRavg = "TTRavg.x") %>%
  mutate(., TTRavg = na_if(TTRavg, 91.00)) #Subset only crabs that have data from the all timepoints
head(modTTRSubset) #Confirm formatting
```

### Demographic variables

```{r}
TTRSubsetDV <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = modTTRSubset, REML = FALSE) #Run model with sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
```

```{r}
TTRSubsetSex <- lmer(log(TTRavg) ~ integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = modTTRSubset, REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRSubsetSex, TTRSubsetDV)
```

```{r}
TTRSubsetIntegument <- lmer(log(TTRavg) ~ as.factor(sex) + carapace.width + weight + as.factor(missing.swimmer) + 
                            (1|crab.ID), 
                          data = modTTRSubset, REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRSubsetIntegument, TTRSubsetDV) #Integument color is not significant
```

```{r}
TTRSubsetCW <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + weight + as.factor(missing.swimmer) + 
                    (1|crab.ID), 
                  data = modTTRSubset, REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRSubsetCW, TTRSubsetDV) #CW is not significant
```

```{r}
TTRSubsetWeight <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      data = modTTRSubset, REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRSubsetWeight, TTRSubsetDV) #Weight is not significant
```

```{r}
TTRSubsetSwimmer <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight +
                         (1|crab.ID), 
                       data = modTTRSubset, REML = FALSE) #Run model without variable of interest to identify significance
anova(TTRSubsetSwimmer, TTRSubsetDV) #Missing swimmer is not significant
```

No demographic variables are significant for the subset.

### Explanatory variables

```{r}
TTRSubsetEV <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                     (1|crab.ID), 
                   data = modTTRSubset, REML = FALSE) #Run model with treatment, timepoint, and their interaction
```

```{r}
TTRSubsetTreatment <- lmer(log(TTRavg) ~ timepoint + 
                           (1|crab.ID), 
                         data = modTTRSubset, REML = FALSE) #Run null model without treatment to identify significance of predictor
anova(TTRSubsetTreatment, TTRSubsetEV) #Treatment is a significant predictor
```

```{r}
TTRSubsetTimepoint <- lmer(log(TTRavg) ~ treatment +
                           (1|crab.ID),
                         data = modTTRSubset, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRSubsetTimepoint, TTRSubsetEV) #Timepoint is a significant predictor
```

```{r}
TTRSubsetInteraction <- lmer(log(TTRavg) ~ treatment + timepoint + 
                             (1|crab.ID),
                           data = modTTRSubset, REML = FALSE) #Run model without interaction to identify significance of predictor. The full model constructed does not include an interaction
anova(TTRSubsetInteraction, TTRSubsetEV) #Interaction is a significant predictor
```

### Genotype variables

```{r}
TTRSubsetfullGenotype <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(genotype) + 
                          (1|crab.ID),
                        data = modTTRSubset, REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRSubsetEV, TTRSubsetfullGenotype) #No significant impact of genotype
```

I could only test the full genotype, since there were only CT and CC crabs left at the end!

### Final model

```{r}
summary(TTRSubsetEV)
```


```{r}
var.lmer <- c(0.06478, 0.46177) #Save variances of random effects as a new vector (crab ID, residual)
percentvar.lmer <- (100*var.lmer)/sum(var.lmer) #Calculate percent variances
percentvar.lmer[1] #Crab ID accounts for 12.30273% of variance in the subset model
```

### Save model output

```{r}
TTRSubsetModelComparisons<- rbind(broom::tidy(anova(TTRSubsetSex, TTRSubsetDV)),
                            broom::tidy(anova(TTRSubsetIntegument, TTRSubsetDV)),
                            broom::tidy(anova(TTRSubsetCW, TTRSubsetDV)),
                            broom::tidy(anova(TTRSubsetWeight, TTRSubsetDV)),
                            broom::tidy(anova(TTRSubsetSwimmer, TTRSubsetDV)),
                            broom::tidy(anova(TTRSubsetTreatment, TTRSubsetEV)),
                            broom::tidy(anova(TTRSubsetTimepoint, TTRSubsetEV)),
                            broom::tidy(anova(TTRSubsetInteraction, TTRSubsetEV)),
                            broom::tidy(anova(TTRSubsetEV, TTRSubsetfullGenotype))) %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Create table for model comparison output. Adjust p-values for multiple comparisons
TTRSubsetModelComparisons
```

```{r}
write.csv(TTRSubsetModelComparisons, "ttr-subset-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save model output
```

# Threshold analysis

## Define threshold

```{r}
modTTR <- modTTR %>%
  mutate(., TTRthresh = case_when(!(is.na(trial.1 & trial.2 & trial.3) == FALSE) ~ 0,
                                  is.na(trial.1 & trial.2 & trial.3) == FALSE ~ 1)) #Create a new column where 1 = righted within 90 seconds in all trials and 0 = failure to right within that threshold
head(modTTR) #Confirm modifications
```

## Assess importance of demographic variables

```{r}
TTRthreshModelDV <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                            (1|crab.ID), 
                          family = binomial(), 
                          data = modTTR) #Create a binomial model, where 1 = success of righting within the defined threshold
```

```{r}
TTRthreshSex <- glmer(I(TTRthresh == 1) ~ integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      family = binomial(), 
                      data = modTTR)
anova(TTRthreshSex, TTRthreshModelDV) #No significant influence of sex
```

```{r}
TTRthreshIntegument <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + carapace.width + weight + as.factor(missing.swimmer) + 
                               (1|crab.ID), 
                             family = binomial(), 
                             data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshIntegument, TTRthreshModelDV) #Integument color not significant
```

```{r}
TTRthreshCW <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + weight + as.factor(missing.swimmer) + 
                       (1|crab.ID), 
                     family = binomial(), 
                     data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshCW, TTRthreshModelDV) #CW not significant
```

```{r}
TTRthreshModelWeight <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + as.factor(missing.swimmer) + 
                                (1|crab.ID), 
                              family = binomial(), 
                              data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshModelWeight, TTRthreshModelDV) #Weight not significant
```

```{r}
TTRthreshModelSwimmer <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + weight +
                                 (1|crab.ID), 
                               family = binomial(), 
                               data = modTTR) #Create a binomial model, where 1 = success of righting within the defined threshold
anova(TTRthreshModelSwimmer, TTRthreshModelDV) #Missing swimmer not significant
```

## Assess importance of explanatory variables

```{r}
TTRthreshModelEV <- glmer(I(TTRthresh == 1) ~ treatment + timepoint + treatment:timepoint +
                            (1|crab.ID), 
                          family = binomial(), 
                          data = modTTR) #Create a binomial model, where 1 = success of righting within the defined threshold
```

```{r}
TTRthreshNullTreatment <- glmer(I(TTRthresh == 1) ~ timepoint +
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTR) #Create null model without treatment
anova(TTRthreshNullTreatment, TTRthreshModelEV) #Compare models
```

```{r}
TTRthreshNullTimepoint <- glmer(I(TTRthresh == 1) ~ treatment +
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTR) #Create null model without timepoint
anova(TTRthreshNullTimepoint, TTRthreshModelEV) #Compare models
```

```{r}
TTRthreshNullInteraction <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                    (1|crab.ID), 
                                  family = binomial(), 
                                  data = modTTR) #Create null model without interaction
anova(TTRthreshNullInteraction, TTRthreshModelEV) #Compare models
```

## Assess importance of genotype

Alright, there are significant impacts of treatment and time on righting, but no impact of their interaction. Now I want to test the impact of genotype and population source on binary righting response. 

```{r}
TTRthreshModelRevised <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                 (1|crab.ID), 
                               family = binomial(), 
                               data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
```

```{r}
TTRthreshGenotype <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                             as.factor(genotype) +
                             (1|crab.ID), 
                           family = binomial(), 
                           data = modTTR) #Create model with genotype
anova(TTRthreshModelRevised, TTRthreshGenotype) #Compare models
```

```{r}
TTRthreshpresenceC <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                              as.factor(presenceC) +
                              (1|crab.ID), 
                            family = binomial(), 
                            data = modTTR) #Create model with presenceC
anova(TTRthreshModelRevised, TTRthreshpresenceC) #Compare models
```

```{r}
TTRthreshpresenceT <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                              as.factor(presenceT) +
                              (1|crab.ID), 
                            family = binomial(), 
                            data = modTTR) #Create model with presenceT
anova(TTRthreshModelRevised, TTRthreshpresenceT) #Compare models
```

```{r}
TTRthreshSource <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                           as.factor(source) +
                           (1|crab.ID), 
                         family = binomial(), 
                         data = modTTR) #Create null model without interaction
anova(TTRthreshModelRevised, TTRthreshSource) #Compare models
```

## Final model

```{r}
summary(TTRthreshModelEV)
```

## Save model output

```{r}
TTRthreshModelComparisons <- rbind(broom::tidy(anova(TTRthreshSex, TTRthreshModelDV)),
                                   broom::tidy(anova(TTRthreshIntegument, TTRthreshModelDV)),
                                   broom::tidy(anova(TTRthreshCW, TTRthreshModelDV)),
                                   broom::tidy(anova(TTRthreshModelWeight, TTRthreshModelDV)),
                                   broom::tidy(anova(TTRthreshModelSwimmer, TTRthreshModelDV)),
                                   broom::tidy(anova(TTRthreshNullTreatment, TTRthreshModelEV)),
                                   broom::tidy(anova(TTRthreshNullTimepoint, TTRthreshModelEV)),
                                   broom::tidy(anova(TTRthreshNullInteraction, TTRthreshModelEV)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshGenotype)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshpresenceC)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshpresenceT)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshSource))) %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Create a table of model comparison output. Adjust p-values.
TTRthreshModelComparisons
```

```{r}
write.csv(TTRthreshModelComparisons, "ttr-binomial-model-comparison-output.csv", quote = FALSE, row.names = FALSE) #Save model comparison table
```

The impacts of presence of the T allele and source population on whether or not a crab righted itself are no longer significant after multiple correction. I'll include genotype-specific information in the final figure since there was a trend.

## Calculate proportions for righting

```{r}
modTTR %>%
  group_by(., treatment, timepoint, TTRthresh) %>%
  summarize(., count = n())
```

```{r}
modTTR %>%
  group_by(., treatment, timepoint, presenceT, TTRthresh) %>%
  filter(., TTRthresh == 0) %>%
  summarize(., count = n())
```

## Sensitivity test

```{r}
modTTRSubset <- modTTRSubset %>%
  mutate(., TTRthresh = case_when(!(is.na(trial.1 & trial.2 & trial.3) == FALSE) ~ 0,
                                  is.na(trial.1 & trial.2 & trial.3) == FALSE ~ 1)) #Create a new column where 1 = righted within 90 seconds in all trials and 0 = failure to right within that threshold
head(modTTRSubset) #Confirm modifications
```

### Demographic variables

```{r}
TTRthreshSubsetModelDV <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                            (1|crab.ID), 
                          family = binomial(), 
                          data = modTTRSubset) #Create a binomial model, where 1 = success of righting within the defined threshold
```

```{r}
TTRthreshSubsetSex <- glmer(I(TTRthresh == 1) ~ integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      family = binomial(), 
                      data = modTTRSubset)
anova(TTRthreshSubsetSex, TTRthreshSubsetModelDV) #No significant influence of sex
```

```{r}
TTRthreshSubsetIntegument <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + carapace.width + weight + as.factor(missing.swimmer) + 
                               (1|crab.ID), 
                             family = binomial(), 
                             data = modTTRSubset) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshSubsetIntegument, TTRthreshSubsetModelDV) #Integument color not significant
```

```{r}
TTRthreshSubsetCW <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + weight + as.factor(missing.swimmer) + 
                       (1|crab.ID), 
                     family = binomial(), 
                     data = modTTRSubset) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshSubsetCW, TTRthreshSubsetModelDV) #CW not significant
```

```{r}
TTRthreshSubsetModelWeight <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + as.factor(missing.swimmer) + 
                                (1|crab.ID), 
                              family = binomial(), 
                              data = modTTRSubset) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
anova(TTRthreshSubsetModelWeight, TTRthreshSubsetModelDV) #Weight not significant
```

```{r}
TTRthreshSubsetModelSwimmer <- glmer(I(TTRthresh == 1) ~ as.factor(sex) + integument.cont + carapace.width + weight +
                                 (1|crab.ID), 
                               family = binomial(), 
                               data = modTTRSubset) #Create a binomial model, where 1 = success of righting within the defined threshold
anova(TTRthreshSubsetModelSwimmer, TTRthreshSubsetModelDV) #Missing swimmer not significant
```

No demographic variables are significant.

### Explanatory variables

```{r}
TTRthreshSubsetEV <- glmer(I(TTRthresh == 1) ~ treatment + timepoint + treatment:timepoint +
                            (1|crab.ID), 
                          family = binomial(), 
                          data = modTTRSubset) #Create a binomial model, where 1 = success of righting within the defined threshold
```

```{r}
TTRthreshSubsetTreatment <- glmer(I(TTRthresh == 1) ~ timepoint +
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTRSubset) #Create null model without treatment
anova(TTRthreshSubsetTreatment, TTRthreshSubsetEV) #Compare models
```

```{r}
TTRthreshSubsetTimepoint <- glmer(I(TTRthresh == 1) ~ treatment +
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTRSubset) #Create null model without timepoint
anova(TTRthreshSubsetTimepoint, TTRthreshSubsetEV) #Compare models
```

```{r}
TTRthreshSubsetInteraction <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                    (1|crab.ID), 
                                  family = binomial(), 
                                  data = modTTRSubset) #Create null model without interaction
anova(TTRthreshSubsetInteraction, TTRthreshSubsetEV) #Compare models
```

### Genotype variables

```{r}
TTRthreshSubsetRevised <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                 (1|crab.ID), 
                               family = binomial(), 
                               data = modTTRSubset) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
```

```{r}
TTRthreshSubsetGenotype <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                             as.factor(genotype) +
                             (1|crab.ID), 
                           family = binomial(), 
                           data = modTTRSubset) #Create model with genotype
anova(TTRthreshSubsetRevised, TTRthreshSubsetGenotype) #Compare models
```

### Save model output

```{r}
TTRthreshSubsetComparisons <- rbind(broom::tidy(anova(TTRthreshSubsetSex, TTRthreshSubsetModelDV)),
                                   broom::tidy(anova(TTRthreshSubsetIntegument, TTRthreshSubsetModelDV)),
                                   broom::tidy(anova(TTRthreshSubsetCW, TTRthreshSubsetModelDV)),
                                   broom::tidy(anova(TTRthreshSubsetModelWeight, TTRthreshSubsetModelDV)),
                                   broom::tidy(anova(TTRthreshSubsetModelSwimmer, TTRthreshSubsetModelDV)),
                                   broom::tidy(anova(TTRthreshSubsetTreatment, TTRthreshSubsetEV)),
                                   broom::tidy(anova(TTRthreshSubsetTimepoint, TTRthreshSubsetEV)),
                                   broom::tidy(anova(TTRthreshSubsetInteraction, TTRthreshSubsetEV)),
                                   broom::tidy(anova(TTRthreshSubsetRevised, TTRthreshSubsetGenotype))) %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Create a table of model comparison output. Adjust p-values.
TTRthreshSubsetComparisons
```

```{r}
write.csv(TTRthreshSubsetComparisons, "ttr-binomial-subset-model-comparison-output.csv", quote = FALSE, row.names = FALSE) #Save model comparison table
```

# Plots

## Average TTR plot

```{r}
modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("0C", "5C"),
                     labels = c("1.7", "5.4")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("0C", "5C"),
                     labels = c("1.7", "5.4")) +
  theme_classic(base_size = 15) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
ggsave("figures/time-to-right-avg-boxplot.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  filter(., treatment == "5C") %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("0C", "5C"),
                     labels = c("1.7", "5.4")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("0C", "5C"),
                     labels = c("1.7", "5.4")) +
  theme_classic(base_size = 15) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
```

## Threshold analysis plot

```{r}
treatment_labels <- c("0C" = "1.5ºC",
                      "5C" = "5ºC") #Modify facet labels for treatment
timepoint_labels <- c("0" = "0",
                      "1" = "4",
                      "2" = "22",
                      "3" = "28",
                      "4" = "46",
                      "5" = "94") #Modify facet labels for timepoint
```

```{r}
modTTR %>%
  group_by(., treatment, timepoint) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(treatment~timepoint, labeller = labeller(treatment = treatment_labels,
                                                      timepoint = timepoint_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips.pdf", height = 8.5, width = 11)

#Take TTR data and group by timepoint. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

```{r}
modTTR %>%
  group_by(., timepoint, treatment) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-timepoint.pdf", height = 8.5, width = 11)
```

```{r}
presenceT_labels <- c("N" = "CC",
                      "Y" = "CT or TT") #Modify facet labels for precense of the T allele
source_labels <- c("EP" = "Eel Pond, MA",
                   "NB" = "New Bedford, MA") #Modify facet labels for source
```

```{r}
modTTR %>%
  group_by(., source, presenceT) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(source~presenceT, labeller = labeller(presenceT = presenceT_labels, 
                                                   source = source_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-sourceT.pdf", height = 8.5, width = 11)

#Take TTR data and group by source and the presence of the T allele. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

Perhaps population source is significant because all of the Eel Pond crabs are CT! I will make a plot showing the number of crabs that didn't flip based on the presence of the T allele.

```{r}
modTTR %>%
  group_by(., timepoint, presenceT) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(~presenceT, labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-timepointT.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  group_by(., presenceT) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(~presenceT, labeller = labeller(presenceT = presenceT_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-trialT.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(., is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE) %>%
  group_by(., timepoint, presenceT) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  ggplot(., aes(x = timepoint, y = count, fill = presenceT)) +
  geom_bar(position = "fill", stat = "identity") + 
  labs(y = "Proportion") +
  scale_x_continuous(name = "Hours",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey70", "grey10"),
                    labels = c("CC", "CT or TT")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-noFlips-timeT.pdf", height = 8.5, width = 11)
#Take TTR data and group by timepoint. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

```{r}
modTTR %>%
  filter(., is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE) %>%
  group_by(., timepoint, presenceT, source) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  unite(., "source_T", source, presenceT) %>%
  ggplot(., aes(x = timepoint, y = count, fill = source_T)) +
  geom_bar(position = "fill", stat = "identity") + 
  labs(y = "Proportion") +
  scale_x_continuous(name = "Hours",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_fill_manual(name = "Population and Genotype",
                    values = c("grey90", "grey45", "grey5"),
                    labels = c("Eel Pond CT",
                               "New Bedford CC",
                               "New Bedford CT or TT")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-noFlips-timeSourceT.pdf", height = 8.5, width = 11)
#Take TTR data and group by timepoint. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

## Multipanel plots

I want to add significance information to the plot. Since treatments were significantly different from eachother at all timepoints except hour 0, I don't think that contrast is as interesting as understanding within-treatment differences by time.

```{r}
time_letters <- cld(TTRmodel_time[[1]], Letters = letters) #Use cld from multcomp package to get letter groupings to highlight which groups are significantly differnt from eachother.
time_letters_plot <- time_letters %>% 
  broom::tidy(.) %>%
  dplyr::rename(timepoint = "timepoint_factor") %>%
  mutate(.group = case_when(treatment == "5C" ~ str_to_upper(.group),
                            treatment == "0C" ~ .group)) %>% 
  arrange(timepoint, treatment) %>% 
  mutate(label_y = c(9, 9,
                     26, 9,
                     89, 36,
                     69, 29,
                     56, 22,
                     59, 16))
#Take lettering and tidy using broom. Rename the timepoint column to match plotting data. Convert the lettering for 5C to uppercase, since 5C and 1C are different from eachother. Arrange by timepoint and treatment. Manually add label height for placing letters above the boxes.
time_letters_plot #Confirm changes
```

```{r}
averageTTRMA <- modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  geom_text(data = time_letters_plot,
            aes(x = timepoint, y = label_y, label = .group, group = treatment, color = treatment),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE, show.legend = FALSE) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 10), seq(20, 100, 20))) +
  scale_x_discrete(name = "",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "",
                     breaks = c("0C", "5C"),
                     labels = c("1.5ºC", "5ºC")) +
  scale_shape_manual(values = c(15, 19),
                     name = "",
                     breaks = c("0C", "5C"),
                     labels = c("1.5ºC", "5ºC")) +
  ggtitle("A. Righting Response for MA Crabs") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"),
                                        strip.text.y = element_blank(),
                                        legend.position = c(0.08, 0.9)) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
averageTTRMA
```

```{r}
averageTTRMAGenotype <- modTTR %>%
  dplyr::select(., c(timepoint, treatment, presenceT, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  mutate(., presenceTtreat = case_when(presenceT == "Y" & treatment == "0C" ~ "1_Y_0C",
                                       presenceT == "N" & treatment == "0C" ~ "2_N_0C",
                                       presenceT == "Y" & treatment == "5C" ~ "3_Y_5C",
                                       presenceT == "N" & treatment == "5C" ~ "4_N_5C")) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = presenceTtreat, shape = presenceTtreat, fill = presenceTtreat)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.5) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 10), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[1],
                                plotColors[2], plotColors[2]),
                     labels = c("1.5ºC, CT or TT",
                                "1.5ºC, CC",
                                "5ºC, CT or TT",
                                "5ºC, CC"),
                     name = "") +
  scale_fill_manual(values = c(plotColors[1], "NA",
                               plotColors[2], "NA"),
                    labels = c("1.5ºC, CT or TT",
                               "1.5ºC, CC",
                               "5ºC, CT or TT",
                               "5ºC, CC"),
                    name = "") +
  scale_shape_manual(values = c(15, 15,
                                19, 19),
                     labels = c("1.5ºC, CT or TT",
                                "1.5ºC, CC",
                                "5ºC, CT or TT",
                                "5ºC, CC"),
                     name = "") +
  ggtitle("B. Righting Response by Genotype") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"),
                                        strip.text.y = element_blank(),
                                        legend.position = c(0.14, 0.8)) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
averageTTRMAGenotype
```

```{r}
failureToRightMA <- modTTR %>%
  group_by(., timepoint, treatment) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion of Crabs") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  ggtitle("C. Proportion Righted") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"), 
                                        legend.position = "right")
failureToRightMA
```

```{r}
genotypeFailureMA <- modTTR %>%
  group_by(., timepoint, presenceT) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(~presenceT, labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  ggtitle("D. Righting by Genotype") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"),
                                        legend.position = "right")
genotypeFailureMA
```

```{r}
averageTTRMA / averageTTRMAGenotype / (failureToRightMA | genotypeFailureMA  + plot_layout(guides = "collect"))
ggsave("figures/multipanel-MA-plot.pdf", width = 8.5, height = 11)
```

# Individual-level analysis

## Q10 analysis

Q10 is a metric used to quantify temperature sensitivity, generally defined as the change in a physiological response over a 10 degree temperature interval. For most organisms, Q10 is around 2 (a two-fold change in a physiological or chemical response due to a 10 degree temperature change).

Since Q10 is usually used for rates, I will apply the Q10 approach to examine differences in righting response for crabs that actually righted themselves.

```{r}
modTTRQ10 <- modTTR %>%
  dplyr::select(., timepoint, crab.ID, TTRavg, treatment, presenceT) %>%
  group_by(., crab.ID) %>%
  pivot_wider(., names_from = timepoint, values_from = TTRavg, names_prefix = "timepoint") %>%
  mutate_all(~ifelse(is.nan(.), 91, .)) %>%
  filter(., is.na(timepoint0) == FALSE) %>%
  filter(., is.na(timepoint1) == FALSE) %>%
  filter(., is.na(timepoint2) == FALSE) %>%
  filter(., is.na(timepoint3) == FALSE) %>%
  filter(., is.na(timepoint4) == FALSE) %>%
  filter(., is.na(timepoint5) == FALSE) %>%
  mutate(temperature = case_when(treatment == "0C" ~ 1.56,
                                 treatment == "5C" ~ 5.39)) %>%
  arrange(., temperature) %>%
  mutate(., timepoint1_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint1, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint2_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint2, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint3_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint3, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint4_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint4, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>% 
  mutate(., timepoint5_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint5, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  dplyr::select(., crab.ID, treatment, presenceT,
         timepoint1_Q10, timepoint2_Q10, timepoint3_Q10, timepoint4_Q10, timepoint5_Q10) %>%
  pivot_longer(., cols = !c(crab.ID, treatment, presenceT), names_to = "timepoint", values_to = "Q10") %>%
  mutate(., timepoint = gsub(x = timepoint, pattern = "_Q10", replacement = "")) %>%
  mutate(., timepoint = gsub(x = timepoint, pattern = "timepoint", replacement = "")) %>%
  mutate(., timepoint = as.numeric(timepoint))
#Select columns of interest. Add columns for presence of C or T alleles. Pivot wider. Change NaN to 91 and remove any NA measurements. Add a column for temperature based on the treatment column, and arrange data by temperature. Use the Q10 function within the respirometry package to calculate Q10 values for timepoint 0 vs. all other timepoints. Select columns of interest and pivot longer. Use a series of gsub to modify timepoint column and convert to a character format that would be organized numerically. Use a similar approach for treatment.
head(modTTRQ10)
length(modTTRQ10$crab.ID %>% unique(.)) #21 crabs
```

```{r}
modTTR %>%
  select(., timepoint, crab.ID, TTRavg, treatment, presenceT) %>%
  group_by(., crab.ID) %>%
  pivot_wider(., names_from = timepoint, values_from = TTRavg, names_prefix = "timepoint") %>%
  mutate_all(~ifelse(is.nan(.), 91, .)) %>%
  filter(., is.na(timepoint0) == FALSE) %>%
  filter(., is.na(timepoint1) == FALSE) %>%
  filter(., is.na(timepoint2) == FALSE) %>%
  filter(., is.na(timepoint3) == FALSE) %>%
  filter(., is.na(timepoint4) == FALSE) %>%
  filter(., is.na(timepoint5) == FALSE) %>%
  mutate(temperature = case_when(treatment == "0C" ~ 1.56,
                                 treatment == "5C" ~ 5.39)) %>%
  arrange(., temperature) %>%
  mutate(., timepoint1_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint1, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint2_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint2, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint3_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint3, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  mutate(., timepoint4_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint4, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>% 
  mutate(., timepoint5_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint5, 
                                 T2 = 14.84, 
                                 T1 = temperature)) %>%
  select(., crab.ID, treatment, presenceT,
         timepoint1_Q10, timepoint2_Q10, timepoint3_Q10, timepoint4_Q10, timepoint5_Q10) %>%
  write.csv("Q10-calculations.csv") #Save wide dataframe as a csv
```


```{r}
modTTRQ10 %>%
  ggplot(., mapping = aes(x = timepoint, y = Q10, color = treatment, shape = treatment, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(. ~ treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_y_continuous(breaks = seq(0, 1, 0.25)) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     guide = "none") +
  scale_shape_manual(values = c(15, 17),
                     guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Plot Q10 data with timepoint on the x-axis, Q10 on the y-axis, and color and shape defined by treatment. Group by crab.ID. Add points and lines connecting values for each crab. Facet by treatment, allow y-axis to vary by each facet, and include facet labels. Add x-axis label and specify x-axis tick labels. Add shape and color values for each treatment but do not include a legend.
ggsave("figures/time-to-right-average-individuals-Q10.pdf", height = 8.5, width = 11)
```

```{r}
modTTRQ10 %>%
  ggplot(., mapping = aes(x = timepoint, y = Q10, color = treatment, shape = treatment, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_grid(presenceT ~ treatment, labeller = labeller(treatment = treatment_labels,
                                                        presenceT = presenceT_labels)) +
  scale_y_continuous(breaks = seq(0, 1, 0.25)) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     guide = "none") +
  scale_shape_manual(values = c(15, 17),
                     guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Plot Q10 data with timepoint on the x-axis, Q10 on the y-axis, and color and shape defined by treatment. Group by crab.ID. Add points and lines connecting values for each crab. Facet by treatment, allow y-axis to vary by each facet, and include facet labels. Add x-axis label and specify x-axis tick labels. Add shape and color values for each treatment but do not include a legend.
ggsave("figures/time-to-right-average-individuals-Q10-genotype.pdf", height = 8.5, width = 11)
```

## Crabs tracked throught the experiment

I tracked certain individuals throughout the course of the experiment, so I want to see how their responses changed over time.

```{r}
left_join(modTTRQ10, modTTR, by = c("crab.ID", "treatment", "presenceT", "timepoint")) %>%
  dplyr::select(., -c(Q10:trial.3, sex:integument.cont, TTRSE:source)) %>%
  filter(., treatment != "0C") %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, group = crab.ID)) +
  geom_point(size = 2, color = plotColors[2]) + geom_line(lty = 3, color = plotColors[2]) +
  facet_wrap(~crab.ID, scales = "free_y") +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = 1:5,
                     labels = c(4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Average Time-to-Right (s)") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Join TTR data with Q10 data to get individuals that were tracked throughout time. Remove superfluous columns and entries at 5ºC. Create point and line graph and wrap by crab ID.
ggsave("figures/time-to-right-individuals-5C.pdf", height = 8.5, width = 11)
```

```{r}
left_join(modTTRQ10, modTTR, by = c("crab.ID", "treatment", "presenceT", "timepoint")) %>%
  dplyr::select(., -c(date:trial.3, sex:TTRSE, genotype:presenceC)) %>%
  filter(., treatment != "5C") %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRthresh, color = presenceT, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(~crab.ID) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = 1:5,
                     labels = c(4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Right?",
                     breaks = c(0, 1),
                     labels = c("No", "Yes")) +
  scale_colour_manual(name = "Genotype",
                      values = c("grey70", "grey10"),
                      labels = c("CC", "CT")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Join TTR data with Q10 data to get individuals that were tracked throughout time. Remove superfluous columns and entries at 5ºC. Create point and line graph and wrap by crab ID.
ggsave("figures/time-to-right-noFlips-individuals-Q10.pdf", height = 8.5, width = 11)
```

```{r}
left_join(modTTRQ10, modTTR, by = c("crab.ID", "treatment", "presenceT", "timepoint")) %>%
  select(., -c(Q10:trial.3, sex:TTRSE, genotype:presenceC)) %>%
  filter(., treatment != "5C") %>%
  pivot_wider(., names_from = "timepoint", names_prefix = "timepoint", values_from = "TTRthresh") %>%
  mutate(., pattern = case_when(timepoint2 == 0 & timepoint3 == 0 & timepoint4 == 0 & timepoint5 == 0 ~ "4",
                                timepoint2 == 1 & timepoint3 == 1 & timepoint4 == 1 & timepoint5 == 1 ~ "1",
                                (timepoint2 == 0 | timepoint3 == 0) & timepoint5 == 1 ~ "2",
                                (timepoint3 == 1 | timepoint4 == 1) & timepoint5 == 0 ~ "3")) %>%
  group_by(., presenceT, pattern) %>%
  summarize(., count = n()) %>%
  ggplot(., mapping = aes(x = presenceT, y = count, fill = pattern)) +
  geom_bar(position = "fill", stat = "identity") +
  labs(y = "Proportion") +
  scale_x_discrete(name = "Genotype",
                   labels = c("CC", "CT")) +
  scale_fill_manual(name = "Pattern",
                    labels = c("Always", "Recovered", "Failed", "Never"),
                    values = c("grey90", "grey60", "grey30", "grey0")) +
  theme_classic(base_size = 15) #Take data and pivot timepoint and threshold information. Assign patterns where 1 = always righting, 2 = recovered, 3 = failure to recover, and 4 = never righting. Group data by genotype and pattern and generate counts. Create stacked barplot.
ggsave("figures/time-to-right-average-individuals-Q10-genotype-patterns.pdf", height = 8.5, width = 11)
```


```{r}
modTTR %>%
  dplyr::select(., crab.ID, treatment, presenceT, timepoint, source, TTRthresh) %>%
  filter(., timepoint != 0) %>%
  filter(., treatment == "0C") %>%
  group_by(., crab.ID) %>%
  arrange(., crab.ID) %>%
  filter(n() > 2) %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRthresh, color = presenceT, shape = source, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(~crab.ID) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = 1:5,
                     labels = c(4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Right?",
                     breaks = c(0, 1),
                     labels = c("No", "Yes")) +
  scale_colour_manual(name = "Genotype",
                      values = c("grey70", "grey10"),
                      labels = c("CC", "CT")) +
  scale_shape_manual(name = "Population",
                     values = c(15, 19),
                     labels = c("Eel Pond, MA", "New Bedford, MA")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-individuals.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  dplyr::select(., crab.ID, treatment, presenceT, timepoint, source, TTRthresh) %>%
  filter(., timepoint != 0) %>%
  filter(., treatment == "0C") %>%
  group_by(., crab.ID) %>%
  arrange(., crab.ID) %>%
  filter(n() > 2) %>%
  pivot_wider(., names_from = "timepoint", names_prefix = "timepoint", values_from = "TTRthresh") %>%
  mutate(., pattern = case_when(timepoint2 == 0 & timepoint3 == 0 & (timepoint4 == 0 | timepoint5 == 0) ~ "4",
                                timepoint2 == 1 & timepoint3 == 1 & timepoint4 == 1 & timepoint5 == 1 ~ "1",
                                (timepoint2 == 0 | timepoint3 == 0) & timepoint5 == 1 ~ "2",
                                (timepoint3 == 1 | timepoint4 == 1) & timepoint5 == 0 ~ "3")) %>%
  filter(., is.na(pattern) == FALSE) %>%
  group_by(., presenceT, pattern) %>%
  summarize(., count = n()) %>%
  ggplot(., mapping = aes(x = presenceT, y = count, fill = pattern)) +
  geom_bar(position = "fill", stat = "identity") +
  labs(y = "Proportion") +
  scale_x_discrete(name = "Genotype",
                   labels = c("CC", "CT")) +
  scale_fill_manual(name = "Pattern",
                    labels = c("Always", "Recovered", "Failed", "Never"),
                    values = c("grey90", "grey60", "grey30", "grey0")) +
  theme_classic(base_size = 15) #Take data and pivot timepoint and threshold information. Assign patterns where 1 = always righting, 2 = recovered, 3 = failure to recover, and 4 = never righting. Group data by genotype and pattern and generate counts. Create stacked barplot.
ggsave("figures/time-to-right-average-individuals-genotype-patterns.pdf", height = 8.5, width = 11)
```
