---
title: "03-TTR-analysis-WA"
author: "Yaamini Venkataraman"
date: '2024-08-20'
output: html_document
---

In this document I'll examine how time-to-right (TTR) varies by temperature and time for WA crabs.

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../../output/WA/03-TTR-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, include=FALSE}
#install.packages("tidyverse")
#install.packages("plotrix")
#install.packages("RColorBrewer")
#install.packages("lme4")
# install.packages("emmeans")
# install.packages("multcompView")
# install.packages("multcomp")
#install.packages("ggpubr")
#install.packages("patchwork")
#install.packages("respirometry")
require(tidyverse)
require(plotrix)
require(RColorBrewer)
require(lme4)
require(emmeans)
require(multcompView)
require(multcomp)
require(ggpubr)
require(patchwork)
require(respirometry)
```

```{r}
sessionInfo()
```

# Import data

```{r}
rawTTR <- read.csv("../../../data/WA/time-to-right-WA-clean.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTR) #Confirm import. Trial data is in seconds.
```

```{r}
crabMetadata <- read.csv("../../../data/WA/crab-metadata-WA.csv", header = TRUE) %>%
  dplyr::select(., c("number", "tank", "carapace.width", "carapace.length")) %>%
  dplyr::rename(crab.ID = "number") #Import metadata. Select ID, treatment, width and length columns. Rename "ID" as "crab.ID"
head(crabMetadata) #Confirm changes
```

```{r}
genotypeData <- read.csv("../../../data/WA/2024-crab-genotype-WA.csv", header = TRUE, na.strings = "") %>%
  dplyr::select(., -c(timepoint:gel.date, notes)) %>%
  rename(., crab.ID = "number") %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeData) #Confirm genotype dataframe creation
```

# Format data

```{r}
modTTR <- rawTTR %>%
  dplyr::select(., -notes) %>%
  left_join(., crabMetadata, by = c("crab.ID", "tank", "carapace.width")) %>%
  filter(., is.na(trial.1) == FALSE) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., treatment = case_when(tank == "T01" ~ "1.5C",
                                  tank == "T02" ~ "1.5C",
                                  tank == "T03" ~ "1.5C",
                                  tank == "T04" ~ "1.5C",
                                  tank == "T05" ~ "5C",
                                  tank == "T06" ~ "5C",
                                  tank == "T07" ~ "5C",
                                  tank == "T08" ~ "5C")) %>%
  mutate(., missing.swimmer = case_when(missing.swimmer == "Y" ~ missing.swimmer,
                                        missing.swimmer == "N" ~ missing.swimmer,
                                        missing.swimmer == "Y " ~ "Y")) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeData %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTR) #Confirm formatting
```

# Data exploration

```{bash}
mkdir figures
```

```{r}
plotColors <- c(brewer.pal(9, "Blues")[9],
                brewer.pal(9, "Blues")[7]) #Create color scheme
```

## Treatment

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = treatment)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "Timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]), 
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("1.5", "5")) +
  theme_classic(base_size = 15) #Plot average TTR and add lm fits without SE bars. Modify x-axis to only show experimental timepoints where TTR was measured. Assign colors to each treatment. Increase base font size.
ggsave("figures/time-to-right-raw.pdf", height = 8.5, width = 11)
```

I'm now going to make plots for various other factors to see if any of these need to be taken into consideration!

## Sex

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = sex)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-sex.pdf", height = 8.5, width = 11)
```

## Tank

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = tank)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-tank.pdf", height = 8.5, width = 11)
```

## Integument color

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = integument.color)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  scale_color_manual(values = c("#66C2A5","#A6D96A","#D9EF8B","#FFFF99","#FDAE61","#F46D43","#D53E4F"),
                     name = "Integument",
                     breaks = c("BG", "G", "YG", "Y", "YO", "O", "RO"),
                     labels = c("BG", "G", "YG", "Y", "YO", "O", "RO")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-integument.pdf", height = 8.5, width = 11)
```

## Missing swimmer

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = as.factor(missing.swimmer))) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-missswim.pdf", height = 8.5, width = 11)
```

## Weight

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = weight)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-weight.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight
#Weight = 32.84938 g ± 7.61794 g
```

## Carapace width

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = carapace.width)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-carapaceWidth.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(timepoint == 0) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE),
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width
#CW = 52.95062 mm ± 4.012173 mm
```

## Genotype

```{r}
modTTR %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = genotype)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTR$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 95, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-genotype.pdf", height = 8.5, width = 11)
```

```{r}
genotypeData %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart.jpg", height = 8.5, width = 11)
```

### Hardy-Weinberg equilibrium

Since I have the genotype data, I want to see if my population is in Hardy-Weinbergy equilibrium. I used [this calculator](https://www.cog-genomics.org/software/stats) to obtain p-values.

```{r}
genSummarized <- genotypeData %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n())
genSummarized
```

Full population p-value: 0.0001102305839834

This significant p-value means that the population is not in HW equilibrium! That means there are external forces acting upon the genotype distribution in this population.

# Assess differences in TTR based on treatment and time

I will use a linear mixed effects model with individual as a random effect to analyze the data, following [this tutorial](https://jontalle.web.engr.illinois.edu/MISC/lme4/bw_LME_tutorial.pdf).

```{r}
hist(modTTR$TTRavg)
hist(x = log(modTTR$TTRavg)) #Log transformation helps normalize data
```

```{r}
TTRmodel <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                   as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                   (1|crab.ID), 
                 data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model with treatment, timepoint, their interaction, sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
summary(TTRmodel) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```
```{r}
var.lmer <- c(0.02357, 0.47326) #Save variances of random effects as a new vector (crab ID, residuals)
percentvar.lmer <- (100*var.lmer)/sum(var.lmer) #Calculate percent variances
percentvar.lmer[1] #Crab ID accounts for 4.744077% of variance
```

Now that I've run the full model, I want to test the importance of treatment, timepoint, and their interaction as predictors. I'll include all other factors in the null model to account for any variation they may contribute.

```{r}
TTRnullTreatment <- lmer(log(TTRavg) ~ timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run null model without treatment to identify significance of predictor
anova(TTRnullTreatment, TTRmodel) #Treatment is a significant predictor
```

```{r}
TTRnullTimepoint <- lmer(log(TTRavg) ~ treatment + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullTreatment, TTRmodel) #Timepoint is a significant predictor
```

```{r}
TTRnullInteraction <- lmer(log(TTRavg) ~ treatment + timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run model without interaction to identify significance of predictor. The full model constructed does not include an interaction
anova(TTRnullInteraction, TTRmodel) #Interaction is a significant predictor
```

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullTreatment, TTRmodel)),
                            broom::tidy(anova(TTRnullTimepoint, TTRmodel)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodel))) #Create table for model comparison output
TTRModelComparisons
```

```{r}
TTRModelComparisons %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) #Adjust p-values
```

## Posthoc tests for significant predictors

There were significant impacts of treatment, timepoint, and their interaction on average righting response! I will investigate these effects using post-hoc tests with `emmeans`.

### Treatment

The first thing I will do is explore pairwise differences between treatment. Since time was also significant, I'll need to look at treatment differences at ech timepoint.

```{r}
TTRmodel_treatment <- emmeans(TTRmodel, pairwise ~ treatment | timepoint, at = list(timepoint = c(0, 1, 2, 3, 4, 5)), adjust = "tukey", data = modTTR) #Use emmeans to obtain pairwise treatment differences at the timepoints listed. 
pairs(TTRmodel_treatment$emmeans) %>% broom::tidy(.) #Use the pairs function to look at pairwise test output. Clean output using broom.
pairs(TTRmodel_treatment$emmeans) %>% 
  broom::tidy(.) %>%
  write.csv("tukey-test-treatment.csv", quote = FALSE, row.names = FALSE) #Save output as a csv.
```

Interactions can also be investigated by looking at differences in linear model slopes. I will estimate simple slopes to understand how average TTR changes over time for each treatment.

```{r}
TTRmodel_treatment_slope <- emtrends(TTRmodel, pairwise ~ treatment, var = "timepoint", data = modTTR) #Obtain simple slopes for the interaction between each temperature and time. Test if the effect of time significantly differs between the two treatments
TTRmodel_treatment_slope$emtrends #Different slopes for each temperature (makes sense because there was an interaction)
TTRmodel_treatment_slope$contrasts #Significance test for the different slopes.
```

### Timepoint

I want to understand how TTR varied by timepoint within different treatments. There is no way to do this with my original model since timepoint was (rightfully) coded as a continuous variable. In order to get pairwise estimates, I will need to refactor time as a categorical variable, rerun the model, and then use `emmeans` on this revised model.

```{r}
modTTR$timepoint_factor <- as.factor(modTTR$timepoint) #Code timepoint as a factor
TTRmodel_factor <- lmer(log(TTRavg) ~ treatment + timepoint_factor + treatment:timepoint_factor + 
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        data = (modTTR %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Run the original model, but use timepoint as a factor
TTRmodel_time <- emmeans(TTRmodel_factor, pairwise ~ timepoint_factor | treatment, adjust = "tukey", data = modTTR) #Rerun emmeans using timepoint as a factor
pairs(TTRmodel_time$emmeans) %>% broom::tidy(.) #Use the pairs function to look at pairwise output. Clean output using broom.
pairs(TTRmodel_time$emmeans) %>%
  broom::tidy(.) %>%
  write.csv("tukey-test-timepoint.csv", quote = FALSE, row.names = FALSE) #Save output as a csv.
```

## Calculate average values

```{r}
modTTR %>%
  group_by(., treatment, timepoint) %>%
  summarize(., mean = mean(TTRavg, na.rm = TRUE), sd = sd(TTRavg, na.rm = TRUE)) %>%
  write.csv("average-TTR-calculations.csv", quote = FALSE, row.names = FALSE)
```

# Effect of genotype

## Impact on average TTR

Now that I know treatment, time, and their interaction significantly impact TTR, I want to add in genotype information.

```{r}
TTRmodelReduced <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with treatment, timepoint, their interaction, sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
summary(TTRmodelReduced) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
TTRfullGenotype <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(genotype) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRfullGenotype) #No significant impact of genotype
```

```{r}
TTRpresenceC <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceC) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRpresenceC) #No significant impact of C allele
```

```{r}
TTRpresenceT <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceT) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRpresenceT) #No significant impact of T allele
```

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullTreatment, TTRmodel)),
                            broom::tidy(anova(TTRnullTimepoint, TTRmodel)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodel)),
                            broom::tidy(anova(TTRmodelReduced, TTRfullGenotype)),
                            broom::tidy(anova(TTRmodelReduced, TTRpresenceC)),
                            broom::tidy(anova(TTRmodelReduced, TTRpresenceT))) #Create table for model comparison output
TTRModelComparisons
```

```{r}
TTRModelComparisons <- TTRModelComparisons %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Adjust p-values
```

```{r}
write.csv(TTRModelComparisons, "ttr-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save model output
```

There is no significant impact of genotype on average TTR. There are some additional factors I want to understand: how genotype impacts whether or not a crab flipped, and how genotype impacted a difference in TTR.

## Threshold analysis

### Define threshold

```{r}
modTTR <- modTTR %>%
  mutate(., TTRthresh = case_when(!(is.na(trial.1 & trial.2 & trial.3) == FALSE) ~ 0,
                                  is.na(trial.1 & trial.2 & trial.3) == FALSE ~ 1)) #Create a new column where 1 = righted within 90 seconds in all trials and 0 = failure to right within that threshold
head(modTTR) #Confirm modifications
```

### Binomial model

```{r}
TTRthreshModel <- glmer(I(TTRthresh == 1) ~ treatment + timepoint + treatment:timepoint +
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        family = binomial(), 
                        data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModel)
```

```{r}
TTRthreshNullTreatment <- glmer(I(TTRthresh == 1) ~ timepoint +
                                  as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTR) #Create null model without treatment
anova(TTRthreshNullTreatment, TTRthreshModel) #Compare models
```

```{r}
TTRthreshNullTimepoint <- glmer(I(TTRthresh == 1) ~ treatment +
                                  as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = modTTR) #Create null model without timepoint
anova(TTRthreshNullTimepoint, TTRthreshModel) #Compare models
```

```{r}
TTRthreshNullInteraction <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                    as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                    (1|crab.ID), 
                                  family = binomial(), 
                                  data = modTTR) #Create null model without interaction
anova(TTRthreshNullInteraction, TTRthreshModel) #Compare models
```

```{r}
TTRthreshModelComparisons <- rbind(broom::tidy(anova(TTRthreshNullTreatment, TTRthreshModel)),
                                   broom::tidy(anova(TTRthreshNullTimepoint, TTRthreshModel)),
                                   broom::tidy(anova(TTRthreshNullInteraction, TTRthreshModel))) #Create a table of model comparison output
TTRthreshModelComparisons
```

```{r}
TTRthreshModelComparisons %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Adjust p-values
```

Alright, there are significant impacts of treatment and time on righting, but no impact of their interaction. So far everything is similar to the MA population. Now I want to test the impact of genotype on binary righting response. 

```{r}
TTRthreshModelRevised <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                 (1|crab.ID), 
                               family = binomial(), 
                               data = modTTR) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelRevised)
```

Before I look at genotype, I want to look at sex. Coyle et al. (2014) found sex-specific differences in failure to right at cold temperatures.

```{r}
TTRthreshNullSex <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                            integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                            as.factor(genotype) +
                            (1|crab.ID), 
                          family = binomial(), 
                          data = modTTR) #Create null model without sex
anova(TTRthreshNullSex, TTRthreshModelRevised) #Compare models. No significant impact of sex on failure to right.
```

```{r}
TTRthreshGenotype <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                             as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                             as.factor(genotype) +
                             (1|crab.ID), 
                           family = binomial(), 
                           data = modTTR) #Create null model without interaction
anova(TTRthreshModelRevised, TTRthreshGenotype) #Compare models
```

```{r}
TTRthreshpresenceC <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                              as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                              as.factor(presenceC) +
                              (1|crab.ID), 
                            family = binomial(), 
                            data = modTTR) #Create null model without interaction
anova(TTRthreshModelRevised, TTRthreshpresenceC) #Compare models
```

```{r}
TTRthreshpresenceT <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                              as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                              as.factor(presenceT) +
                              (1|crab.ID), 
                            family = binomial(), 
                            data = modTTR) #Create null model without interaction
anova(TTRthreshModelRevised, TTRthreshpresenceT) #Compare models
```

```{r}
TTRthreshModelComparisons <- rbind(broom::tidy(anova(TTRthreshNullTreatment, TTRthreshModel)),
                                   broom::tidy(anova(TTRthreshNullTimepoint, TTRthreshModel)),
                                   broom::tidy(anova(TTRthreshNullInteraction, TTRthreshModel)),
                                   broom::tidy(anova(TTRthreshNullSex, TTRthreshModelRevised)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshGenotype)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshpresenceC)),
                                   broom::tidy(anova(TTRthreshModelRevised, TTRthreshpresenceT))) #Create a table of model comparison output
TTRthreshModelComparisons
```

```{r}
TTRthreshModelComparisons <- TTRthreshModelComparisons %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Adjust p-values
```

```{r}
write.csv(TTRthreshModelComparisons, "ttr-binomial-model-comparison-output.csv", quote = FALSE, row.names = FALSE) #Save model comparison table
```

No significant impact of genotype on whether or not a crab righted!

### Calculate propportions

```{r}
modTTR %>%
  group_by(., treatment, timepoint, TTRthresh) %>%
  summarize(., count = n())
```


# Plots

## Average TTR plot

```{r}
modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  theme_classic(base_size = 15) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
ggsave("figures/time-to-right-avg-boxplot.pdf", height = 8.5, width = 11)
```

## Threshold analysis plot

```{r}
treatment_labels <- c("1.5C" = "1.5ºC",
                      "5C" = "5ºC") #Modify facet labels for treatment
timepoint_labels <- c("0" = "0",
                      "1" = "4",
                      "2" = "22",
                      "3" = "28",
                      "4" = "46",
                      "5" = "94") #Modify facet labels for timepoint
```

```{r}
modTTR %>%
  group_by(., treatment, timepoint) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(treatment~timepoint, labeller = labeller(treatment = treatment_labels,
                                                      timepoint = timepoint_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips.pdf", height = 8.5, width = 11)

#Take TTR data and group by timepoint. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

```{r}
modTTR %>%
  group_by(., timepoint, treatment) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-timepoint.pdf", height = 8.5, width = 11)
```

```{r}
presenceT_labels <- c("N" = "CC",
                      "Y" = "CT or TT") #Modify facet labels for precense of the T allele
```

```{r}
modTTR %>%
  group_by(., treatment, genotype) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(treatment~genotype, labeller = labeller(treatment = treatment_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-tempGenotype.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  group_by(., treatment, presenceT) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(treatment~presenceT, labeller = labeller(treatment = treatment_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-temp-presenceT.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., timepoint, presenceT) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(~presenceT, labeller = labeller(presenceT = presenceT_labels)) +
  scale_x_continuous(name = "Hours",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-timepointT.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., presenceT) %>%
  summarize(., total = n(), 
            trial1No = sum(is.na(trial.1) == TRUE),
            trial2No = sum(is.na(trial.2) == TRUE),
            trial3No = sum(is.na(trial.3) == TRUE)) %>%
  mutate(., trial1Yes = total - trial1No,
         trial2Yes = total - trial2No,
         trial3Yes = total - trial3No) %>%
  pivot_longer(., cols = c(trial1No, trial1Yes,
                           trial2No, trial2Yes,
                           trial3No, trial3Yes), 
               values_to = "count", names_to = "condition") %>%
  mutate(., trial = case_when(condition == "trial1No" ~ 1,
                              condition == "trial1Yes" ~ 1,
                              condition == "trial2No" ~ 2,
                              condition == "trial2Yes" ~ 2,
                              condition == "trial3No" ~ 3,
                              condition == "trial3Yes" ~ 3)) %>%
  mutate(., condition = gsub(pattern = "trial1", replacement = "", x = condition),
         condition = gsub(pattern = "trial2", replacement = "", x = condition),
         condition = gsub(pattern = "trial3", replacement = "", x = condition)) %>%
  ggplot(., aes(x = trial, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_grid(~presenceT, labeller = labeller(presenceT = presenceT_labels)) +
  labs(x = "Trial", y = "Proportion") +
  scale_fill_manual(name = "",
                    values = c("grey70", "grey10")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-trialT.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  filter(., is.na(genotype) == FALSE) %>%
  filter(., is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE) %>%
  group_by(., timepoint, genotype) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  ggplot(., aes(x = timepoint, y = count, fill = genotype)) +
  geom_bar(position = "fill", stat = "identity") + 
  labs(y = "Proportion") +
  scale_x_continuous(name = "Hours",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0"),
                    labels = c("CC", "CT","TT")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-noFlips-timeT.pdf", height = 8.5, width = 11)
#Take TTR data and group by timepoint. Count the number of crabs that did not right within 90 seconds during each trial. Subtract that value from the total to get the number of crabs that did right within 90 seconds. Pivot the data longer. Create a trial column based on the condition column. Remove trial information from the text in the condition column. Pipe modified data into ggplot to create a stacked barplot. Facet by treatment and timepoint using facet labels. Relabel axes. Modify fill scale. Use specific base size with classic theme. For facet labels, use a white border (no border).
```

## Multipanel plot

I want to add significance information to the plot. Since treatments were significantly different from eachother at all timepoints except hour 0, I don't think that contrast is as interesting as understanding within-treatment differences by time.

```{r}
time_letters <- cld(TTRmodel_time[[1]], Letters = letters) #Use cld from multcomp package to get letter groupings to highlight which groups are significantly differnt from eachother.
time_letters_plot <- time_letters %>% 
  broom::tidy(.) %>%
  dplyr::rename(timepoint = "timepoint_factor") %>%
  mutate(.group = case_when(treatment == "5C" ~ str_to_upper(.group),
                            treatment == "1.5C" ~ .group)) %>% 
  arrange(timepoint, treatment) %>% 
  mutate(label_y = c(13, 13, 
                     13, 13, 
                     45, 30, 
                     85, 18, 
                     90, 26, 
                     90, 18))
#Take lettering and tidy using broom. Rename the timepoint column to match plotting data. Convert the lettering for 5C to uppercase, since 5C and 1C are different from eachother. Arrange by timepoint and treatment. Manually add label height for placing letters above the boxes.
time_letters_plot #Confirm changes
```

```{r}
averageTTRWA <- modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  geom_text(data = time_letters_plot,
            aes(x = timepoint, y = label_y, label = .group, group = treatment, color = treatment),
            position = position_dodge(width = 0.75),
            inherit.aes = FALSE, show.legend = FALSE) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("1.5", "5")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("1.5", "5")) +
  ggtitle("A. Average Righting Response for WA Crabs") +
  theme_classic(base_size = 15) + theme(legend.position = "bottom") #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
averageTTRWA
```

```{r}
failureToRightWA <- modTTR %>%
  group_by(., timepoint, treatment) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion of Crabs") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  ggtitle("B. Proportion of WA Crabs that Righted") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"),
                                        legend.position = "bottom")
```

```{r}
averageTTRWA / failureToRightWA
ggsave("figures/multipanel-WA-plot.pdf", height = 11, width = 8.5)
```

# Multipanel genotype plot

## Experiment 1: WA

### Import and format data

```{r}
rawTTRWAHeat <- read.csv("../../../data/mult-pop-heat/WA_HSP_TTR.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRWAHeat)
```

```{r}
genotypeWAHeat <- read.csv("../../../data/mult-pop-heat/WA_genotype.csv", header = TRUE) %>%
  dplyr::select(., -c(extraction.date:gel.date, notes)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeWAHeat) #Confirm import
```

```{r}
modTTRWAHeat <- rawTTRWAHeat %>%
  dplyr::select(., -notes) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeWAHeat %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTRWAHeat) #Confirm formatting
```

### Create plot

```{r}
experiment1WA <- modTTRWAHeat %>%
  filter(., timepoint == 22) %>%
  dplyr::select(., crab.ID, genotype) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y", start = 0) +
  geom_text(aes(label = paste0(round((count/sum(count)*100), digits = 0), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 6,
            color = c("black", "black", "white")) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  ggtitle("A. Experiment 1: WA") +
  theme_void(base_size = 15) + theme(legend.position = "none",
                                     plot.title = element_text(hjust = 0.5)) #Filter for final timepoint and select columns of interest. Group by genotype then create pie chart. Add labels for percentage genotype in population.
experiment1WA
```

## Experiment 1: ME

### Import and format data

```{r}
rawTTRMEHeat <- read.csv("../../../data/mult-pop-heat/ME_HSP_TTR.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRMEHeat)
```

```{r}
genotypeMEHeat <- read.csv("../../../data/mult-pop-heat/ME_genotype.csv", header = TRUE) %>%
  dplyr::select(., -c(extraction.date:gel.date, notes)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeMEHeat) #Confirm import
```

```{r}
modTTRMEHeat <- rawTTRMEHeat %>%
  dplyr::select(., -notes) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeMEHeat %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTRMEHeat) #Confirm formatting
```

### Create plot

```{r}
experiment1ME <- modTTRMEHeat %>%
  filter(., timepoint == 22) %>%
  dplyr::select(., crab.ID, genotype) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y", start = 0) +
  geom_text(aes(label = paste0(round((count/sum(count)*100), digits = 0), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 6,
            color = c("black", "black")) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  ggtitle("A. Experiment 1: ME") +
  theme_void(base_size = 15) + theme(legend.position = "none",
                                     plot.title = element_text(hjust = 0.5)) #Filter for final timepoint and select columns of interest. Group by genotype then create pie chart. Add labels for percentage genotype in population.
experiment1ME
```

## Experiment 2

### Import and format data

```{r}
rawTTRRepeat <- read.csv("../../../data/repeat-heat/time-to-right-genotype-julia.csv", header = TRUE) #Import raw data
head(rawTTRRepeat) #Confirm import. Trial data is in seconds.
```

```{r}
modTTRRepeat <- rawTTRRepeat %>%
  dplyr::select(., -c(notes, probe.number)) %>%
  mutate(., day = case_when(date == "7/31/23" ~ 1,
                            date == "8/1/23" ~ 2,
                            date == "8/2/23" ~ 2,
                            date == "8/4/23" ~ 5)) %>%
  mutate(., treatment = case_when (tank == "10" | tank == "12" ~ "No Pulse",
                                   tank == "11" | tank == "13" ~ "Pulse")) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  filter(., is.na(TTRSE) == FALSE) %>%
  group_by(., day, treatment) %>%
  mutate(., TTRavgFull = mean(TTRavg)) %>%
  mutate(., TTRSEFull = std.error(TTRavg)) %>%
  mutate(., TTRavgFullLow = TTRavgFull - TTRSEFull) %>%
  mutate(., TTRavgFullHigh = TTRavgFull + TTRSEFull) %>%
  ungroup(.) #Remove sampling ID and notes columns. Add new column with day information, combining 8/1 and 8/2 into the same day4. Create a a new column that with an index for width/length. Create new column as a binary for whether or not crabs are missing legs. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations, and remove rows where TTRavg | TTRSE = NA. Calculate average and SE for all samples in a treatment for each day. Add/subtract TTRSEFull to/from TTRavgFull to get bounds. Ungroup.
head(modTTRRepeat) #Confirm formatting
```

### Create plot

```{r}
experiment2Plot <- modTTRRepeat %>%
  filter(., day == 1) %>%
  dplyr::select(., crab.ID, genotype) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y", start = 0) +
  geom_text(aes(label = paste0(round((count/sum(count)*100), digits = 0), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 6,
            color = c("black", "black", "white")) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  ggtitle("C. Experiment 2: WA") +
  theme_void(base_size = 15) + theme(legend.position = "none",
                                     plot.title = element_text(hjust = 0.5)) #rbind data from experiment 1. Filter for final timepoint and select columns of interest. Group by population and genotype then create pie charts. Use position = "fill" to deal with uneven sample sizes. Facet by population and add scale and title information.
experiment2Plot
```

## Experiment 3: MA

### Import and format data

```{r}
genotypeDataMA <- read.csv("../../../data/MA/2024-crab-genotype-MA.csv", header = TRUE, na.strings = "") %>%
  select(., -c(extraction.date:gel.date, notes:QC.gen)) %>%
  mutate(., crab.ID = gsub(pattern = "MA-", replacement = "", x = ID)) %>%
  mutate(., crab.ID = as.numeric(crab.ID)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeDataMA) #Confirm genotype dataframe creation
```

### Create plot

```{r}
experiment3MA <- genotypeDataMA %>%
  filter(., is.na(genotype) == FALSE) %>%
  dplyr::select(., crab.ID, genotype) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y", start = 0) +
  geom_text(aes(label = paste0(round((count/sum(count)*100), digits = 0), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 6,
            color = c("black", "black", "white")) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  ggtitle("D. Experiment 3: MA") +
  theme_void(base_size = 15) + theme(legend.position = "none",
                                     plot.title = element_text(hjust = 0.5)) #rbind data from experiment 1. Filter for final timepoint and select columns of interest. Group by population and genotype then create pie charts. Use position = "fill" to deal with uneven sample sizes. Facet by population and add scale and title information.
experiment3MA
```

## Experiment 3: WA

This data has already been imported in the script.

### Create plot

```{r}
experiment3WA <- genotypeData %>%
  filter(., is.na(genotype) == FALSE) %>%
  dplyr::select(., crab.ID, genotype) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y", start = 0) +
  geom_text(aes(label = paste0(round((count/sum(count)*100), digits = 0), "%")), 
            position = position_stack(vjust = 0.5), 
            size = 6,
            color = c("black", "black", "white")) +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  ggtitle("E. Experiment 3: WA") +
  theme_void(base_size = 15) + theme(legend.position = "right",
                                     plot.title = element_text(hjust = 0.5)) #rbind data from experiment 1. Filter for final timepoint and select columns of interest. Group by population and genotype then create pie charts. Use position = "fill" to deal with uneven sample sizes. Facet by population and add scale and title information.
experiment3WA
```
## Create multipanel plot

```{r}
(experiment1WA | experiment1ME | experiment2Plot) / ( experiment3MA | experiment3WA)
ggsave("figures/multipanel-genotype-plot.pdf", height = 8.5, width = 11)
```

# Individual-level analysis

## Q10 analysis

Q10 is a metric used to quantify temperature sensitivity, generally defined as the change in a physiological response over a 10 degree temperature interval. For most organisms, Q10 is around 2 (a two-fold change in a physiological or chemical response due to a 10 degree temperature change).

```{r}
modTTRQ10 <- modTTR %>%
  select(., timepoint, crab.ID, TTRavg, treatment, genotype) %>%
  group_by(., crab.ID) %>%
  pivot_wider(., names_from = timepoint, values_from = TTRavg, names_prefix = "timepoint") %>%
  mutate_all(~ifelse(is.nan(.), 91, .)) %>%
  filter(., is.na(timepoint0) == FALSE) %>%
  filter(., is.na(timepoint1) == FALSE) %>%
  filter(., is.na(timepoint2) == FALSE) %>%
  filter(., is.na(timepoint3) == FALSE) %>%
  filter(., is.na(timepoint4) == FALSE) %>%
  filter(., is.na(timepoint5) == FALSE) %>%
  mutate(temperature = case_when(treatment == "1.5C" ~ 2.13,
                                 treatment == "5C" ~ 5.19)) %>%
  arrange(., temperature) %>%
  mutate(., timepoint1_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint1, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint2_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint2, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint3_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint3, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint4_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint4, 
                                 T2 = 15, 
                                 T1 = temperature)) %>% 
  mutate(., timepoint5_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint5, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  select(., crab.ID, treatment, genotype,
         timepoint1_Q10, timepoint2_Q10, timepoint3_Q10, timepoint4_Q10, timepoint5_Q10) %>%
  pivot_longer(., cols = !c(crab.ID, treatment, genotype), names_to = "timepoint", values_to = "Q10") %>%
  mutate(., timepoint = gsub(x = timepoint, pattern = "_Q10", replacement = "")) %>%
  mutate(., timepoint = gsub(x = timepoint, pattern = "timepoint", replacement = "")) %>%
  mutate(., timepoint = as.numeric(timepoint))
#Select columns of interest. Add columns for presence of C or T alleles. Pivot wider. Change NaN to 91 and remove any NA measurements. Add a column for temperature based on the treatment column, and arrange data by temperature. Use the Q10 function within the respirometry package to calculate Q10 values for timepoint 0 vs. all other timepoints. Select columns of interest and pivot longer. Use a series of gsub to modify timepoint column and convert to a character format that would be organized numerically. Use a similar approach for treatment.
head(modTTRQ10)
length(modTTRQ10$crab.ID %>% unique(.)) #28 crabs
```

```{r}
modTTR %>%
  select(., timepoint, crab.ID, TTRavg, treatment, genotype) %>%
  group_by(., crab.ID) %>%
  pivot_wider(., names_from = timepoint, values_from = TTRavg, names_prefix = "timepoint") %>%
  mutate_all(~ifelse(is.nan(.), 91, .)) %>%
  filter(., is.na(timepoint0) == FALSE) %>%
  filter(., is.na(timepoint1) == FALSE) %>%
  filter(., is.na(timepoint2) == FALSE) %>%
  filter(., is.na(timepoint3) == FALSE) %>%
  filter(., is.na(timepoint4) == FALSE) %>%
  filter(., is.na(timepoint5) == FALSE) %>%
  mutate(temperature = case_when(treatment == "1.5C" ~ 2.13,
                                 treatment == "5C" ~ 5.19)) %>%
  arrange(., temperature) %>%
  mutate(., timepoint1_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint1, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint2_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint2, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint3_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint3, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  mutate(., timepoint4_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint4, 
                                 T2 = 15, 
                                 T1 = temperature)) %>% 
  mutate(., timepoint5_Q10 = Q10(R2 = timepoint0, 
                                 R1 = timepoint5, 
                                 T2 = 15, 
                                 T1 = temperature)) %>%
  write.csv("Q10-calculations.csv") #Save wider table as a csv
```

```{r}
modTTRQ10 %>%
  ggplot(., mapping = aes(x = timepoint, y = Q10, color = treatment, shape = treatment, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(. ~ treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     guide = "none") +
  scale_shape_manual(values = c(15, 17),
                     guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Plot Q10 data with timepoint on the x-axis, Q10 on the y-axis, and color and shape defined by treatment. Group by crab.ID. Add points and lines connecting values for each crab. Facet by treatment, allow y-axis to vary by each facet, and include facet labels. Add x-axis label and specify x-axis tick labels. Add shape and color values for each treatment but do not include a legend.
ggsave("figures/time-to-right-average-individuals-Q10.pdf", height = 8.5, width = 11)
```

```{r}
modTTRQ10 %>%
  ggplot(., mapping = aes(x = timepoint, y = Q10, color = treatment, shape = treatment, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_grid(genotype ~ treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     guide = "none") +
  scale_shape_manual(values = c(15, 17),
                     guide = "none") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Plot Q10 data with timepoint on the x-axis, Q10 on the y-axis, and color and shape defined by treatment. Group by crab.ID. Add points and lines connecting values for each crab. Facet by treatment, allow y-axis to vary by each facet, and include facet labels. Add x-axis label and specify x-axis tick labels. Add shape and color values for each treatment but do not include a legend.
ggsave("figures/time-to-right-average-individuals-Q10-genotype.pdf", height = 8.5, width = 11)
```

## Crabs tracked throught the experiment

I tracked certain individuals throughout the course of the experiment, so I want to see how their responses changed over time.

```{r}
left_join(modTTRQ10, modTTR, by = c("crab.ID", "treatment", "genotype", "timepoint")) %>%
  select(., -c(date:trial.3, sex:presenceT)) %>%
  filter(., treatment != "5C") %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRthresh, color = genotype, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(~crab.ID) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = 1:5,
                     labels = c(4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Right?",
                     breaks = c(0, 1),
                     labels = c("No", "Yes")) +
  scale_colour_manual(name = "Genotype",
                      values = c("grey70", "grey10"),
                      labels = c("CT", "TT")) +
  scale_shape_manual(name = "Genotype",
                     values = c(15, 19),
                     labels = c("CT", "TT")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white")) #Join TTR data with Q10 data to get individuals that were tracked throughout time. Remove superfluous columns and entries at 5ºC. Create point and line graph and wrap by crab ID.
ggsave("figures/time-to-right-noFlips-individuals-Q10.pdf", height = 8.5, width = 11)
```

```{r}
left_join(modTTRQ10, modTTR, by = c("crab.ID", "treatment", "genotype", "timepoint")) %>%
  select(., -c(Q10:trial.3, sex:presenceT)) %>%
  filter(., treatment != "5C") %>%
  pivot_wider(., names_from = "timepoint", names_prefix = "timepoint", values_from = "TTRthresh") %>%
  mutate(., pattern = case_when(timepoint2 == 0 & timepoint3 == 0 & timepoint4 == 0 & timepoint5 == 0 ~ "4",
                                timepoint2 == 1 & timepoint3 == 1 & timepoint4 == 1 & timepoint5 == 1 ~ "1",
                                (timepoint2 == 0 | timepoint3 == 0 | timepoint4 == 0) & timepoint5 == 1 ~ "2",
                                (timepoint3 == 1 | timepoint4 == 1) & timepoint5 == 0 ~ "3")) %>%
  group_by(., genotype, pattern) %>%
  summarize(., count = n()) %>%
  ggplot(., mapping = aes(x = genotype, y = count, fill = pattern)) +
  geom_bar(position = "fill", stat = "identity") +
  labs(y = "Proportion") +
  scale_x_discrete(name = "Genotype",
                   labels = c("CT", "TT")) +
  scale_fill_manual(name = "Pattern",
                    labels = c("Always", "Recovered", "Failed", "Never"),
                    values = c("grey90", "grey60", "grey30", "grey0")) +
  theme_classic(base_size = 15) #Take data and pivot timepoint and threshold information. Assign patterns where 1 = always righting, 2 = recovered, 3 = failure to recover, and 4 = never righting. Group data by genotype and pattern and generate counts. Create stacked barplot.
ggsave("figures/time-to-right-average-individuals-Q10-genotype-patterns.pdf", height = 8.5, width = 11)
```


```{r}
modTTR %>%
  filter(., is.na(genotype) == FALSE) %>%
  dplyr::select(., crab.ID, treatment, genotype, timepoint, TTRthresh) %>%
  filter(., timepoint != 0) %>%
  filter(., treatment == "1.5C") %>%
  group_by(., crab.ID) %>%
  arrange(., crab.ID) %>%
  filter(n() > 3) %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRthresh, color = genotype, shape = genotype, group = crab.ID)) +
  geom_point(size = 2) + geom_line(lty = 3) +
  facet_wrap(~crab.ID) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = 1:5,
                     labels = c(4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Right?",
                     breaks = c(0, 1),
                     labels = c("No", "Yes")) +
  scale_colour_manual(name = "Genotype",
                      values = c("grey90", "grey70", "grey10"),
                      labels = c("CC", "CT", "TT")) +
  scale_shape_manual(name = "Genotype",
                     values = c(17, 15, 19),
                     labels = c("CC", "CT", "TT")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"))
ggsave("figures/time-to-right-noFlips-individuals.pdf", height = 8.5, width = 11)
```

```{r}
modTTR %>%
  dplyr::select(., crab.ID, treatment, genotype, timepoint, TTRthresh) %>%
  filter(., timepoint != 0) %>%
  filter(., treatment == "1.5C") %>%
  group_by(., crab.ID) %>%
  arrange(., crab.ID) %>%
  filter(n() > 3) %>%
  pivot_wider(., names_from = "timepoint", names_prefix = "timepoint", values_from = "TTRthresh") %>%
  mutate(., pattern = case_when(timepoint2 == 0 & timepoint3 == 0 & timepoint4 == 0 & timepoint5 == 0 ~ "4",
                                timepoint2 == 1 & timepoint3 == 1 & timepoint4 == 1 & timepoint5 == 1 ~ "1",
                                (timepoint2 == 0 | timepoint3 == 0 | timepoint4 == 0) & timepoint5 == 1 ~ "2",
                                (timepoint3 == 1 | timepoint4 == 1) & timepoint5 == 0 ~ "3")) %>%
  filter(., is.na(pattern) == FALSE) %>%
  group_by(., genotype, pattern) %>%
  summarize(., count = n()) %>%
  ggplot(., mapping = aes(x = genotype, y = count, fill = pattern)) +
  geom_bar(position = "fill", stat = "identity") +
  labs(y = "Proportion") +
  scale_x_discrete(name = "Genotype",
                   labels = c("CT", "TT")) +
  scale_fill_manual(name = "Pattern",
                    labels = c("Always", "Recovered", "Failed", "Never"),
                    values = c("grey90", "grey60", "grey30", "grey0")) +
  theme_classic(base_size = 15) #Take data and pivot timepoint and threshold information. Assign patterns where 1 = always righting, 2 = recovered, 3 = failure to recover, and 4 = never righting. Group data by genotype and pattern and generate counts. Create stacked barplot.
ggsave("figures/time-to-right-average-individuals-genotype-patterns.pdf", height = 8.5, width = 11)
```

# Population comparison

The last thing I want to do is compare WA and MA responses!

## Import data

### WA

```{r}
modTTRWA <- modTTR #Save dataframe with WA-specific tag.
modTTRWA$population <- "WA" #Add column with population information
head(modTTRWA)
```

### MA 

```{r}
rawTTRMA <- read.csv("../../../data/MA/time-to-right-MA-clean.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRMA) #Confirm import. Trial data is in seconds.
```

```{r}
crabMetadataMA <- read.csv("../../../data/MA/crab-metadata-MA.csv", header = TRUE) %>%
  dplyr::select(., c("number", "tank", "carapace.width", "carapace.length")) %>%
  dplyr::rename(crab.ID = "number") #Import metadata. Select ID, treatment, width and length columns. Rename "ID" as "crab.ID"
head(crabMetadataMA) #Confirm changes
```

```{r}
genotypeDataMA <- read.csv("../../../data/MA/2024-crab-genotype-MA.csv", header = TRUE, na.strings = "") %>%
  select(., -c(extraction.date:gel.date, notes:QC.gen)) %>%
  mutate(., crab.ID = gsub(pattern = "MA-", replacement = "", x = ID)) %>%
  mutate(., crab.ID = as.numeric(crab.ID)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeDataMA) #Confirm genotype dataframe creation
```

```{r}
modTTRMA <- rawTTRMA %>%
  dplyr::select(., -notes) %>%
  left_join(., crabMetadataMA, by = c("crab.ID", "tank", "carapace.width")) %>%
  filter(., is.na(trial.1) == FALSE) %>%
  mutate(., integument.cont = case_when(integument.color == "B/G" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "Y/G" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "Y/O" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "R/O" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., treatment = case_when(tank == "T01" ~ "0C",
                                  tank == "T02" ~ "0C",
                                  tank == "T03" ~ "0C",
                                  tank == "T05" ~ "5C",
                                  tank == "T06" ~ "5C",
                                  tank == "T07" ~ "5C")) %>%
  mutate(., missing.swimmer = case_when(missing.swimmer == "Y" ~ missing.swimmer,
                                        missing.swimmer == "N" ~ missing.swimmer,
                                        missing.swimmer == "Y " ~ "Y")) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeDataMA %>% dplyr::select(genotype:presenceT)), by = "crab.ID") %>%
  ungroup(.) %>% 
  mutate(., TTRthresh = case_when(!(is.na(trial.1 & trial.2 & trial.3) == FALSE) ~ 0,
                                  is.na(trial.1 & trial.2 & trial.3) == FALSE ~ 1))
#Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data. Create a new column where 1 = righted within 90 seconds in all trials and 0 = failure to right within that threshold. Subset data from hours 0, 22, and 94.
modTTRMA$population <- "MA" #Add column with population information
head(modTTRMA) #Confirm formatting
```

## Format data

```{r}
modTTRCombined <- rbind(modTTRMA, modTTRWA) %>%
  dplyr::select(., treatment, timepoint, crab.ID, 
                TTRavg, TTRthresh, 
                population, 
                sex, integument.cont, carapace.width, weight, missing.swimmer, genotype, presenceC, presenceT) #Rbind data from both populations, and select columns of interest to use in the model
```

## Run model

### Average TTR

```{r}
TTRCombinedNull <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        data = (modTTRCombined %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Null model without population
summary(TTRCombinedNull) #Crab ID doesn't explain any variation!
```

```{r}
TTRPopulation <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                        as.factor(population) +
                        as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      data = (modTTRCombined %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Model with population
anova(TTRCombinedNull, TTRPopulation) #Population is not a significant predictor of avg TTR responses!
```

Population is not a predictor of average TTR responses! 

### Threshold analysis

Next, I want to investigate the impact of population on failure to right.

```{r}
TTRthreshModelCombined <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                  as.factor(population) +
                                  as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                  (1|crab.ID), 
                                family = binomial(), 
                                data = (modTTRCombined %>% filter(., is.na(genotype) == FALSE))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined)
```

```{r}
TTRthreshNullPopulation <- glmer(I(TTRthresh == 1) ~ treatment + timepoint +
                                   as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                                   (1|crab.ID), 
                                 family = binomial(), 
                                 data = (modTTRCombined %>% filter(., is.na(genotype) == FALSE))) #Create null model without treatment
anova(TTRthreshNullPopulation, TTRthreshModelCombined) #Population is not a factor in failure to right!
```

It's interesting that overall, population does not significantly impact failure to right. This seems...odd too me. I'm going to subset the data to the lower temperature and specific timepoints and rerun the model to continue my investigation. I think this approach makes sense because we know that in both populations, time and temperature significantly impacts failure to right.

### Specific threshold analysis

Since I'm looking into specific temperatures and timepoints, there are no repeated measures for individual crabs. I don't need to use a mixed-effects models!

```{r}
modTTRCombinedColder <- modTTRCombined %>%
  filter(., treatment == "0C" | treatment == "1.5C") #Subset data from colder treatments where failure to right was recorded. 0C = MA, 1.5C = WA
head(modTTRCombinedColder) #Confirm subsetting
```

#### 0 hours (`timepoint == 0`)

```{r}
TTRthreshModelCombined0 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 0))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined0) #No impact of population on failure to right at hour 0
```

#### 4 hours (`timepoint == 1`)

```{r}
TTRthreshModelCombined1 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 1))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined1) #No impact of population on failure to right at hour 4
```

#### 22 hours (`timepoint == 2`)

```{r}
TTRthreshModelCombined2 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 2))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined2) #Significant impact of population on failure to right at hour 22
```

#### 28 hours (`timepoint == 3`)

```{r}
TTRthreshModelCombined3 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 3))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined3) #Significant impact of population on failure to right at hour 28
```

#### 46 hours (`timepoint == 4`)

```{r}
TTRthreshModelCombined4 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 4))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined4) #Significant impact of population on failure to right at hour 46
```

#### 94 hours (`timepoint == 5`)

```{r}
TTRthreshModelCombined5 <- glm(I(TTRthresh == 1) ~ as.factor(population) +
                                 as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer),
                               family = binomial(), 
                               data = (modTTRCombinedColder %>% filter(., timepoint == 5))) #Create a binomial model, where 1 = success of righting wihtin the defined threshold
summary(TTRthreshModelCombined5) #Marginal impact of population on failure to right at the end of the experiment
```

#### Correct for multiple comparisons

Since I've run multiple models, I need to correct for multiple comparisons!

```{r}
rbind(broom::tidy(TTRthreshModelCombined0) %>%
        mutate(timepoint = 0),
      broom::tidy(TTRthreshModelCombined1) %>%
        mutate(timepoint = 1),
      broom::tidy(TTRthreshModelCombined2) %>%
        mutate(timepoint = 2),
      broom::tidy(TTRthreshModelCombined3) %>%
        mutate(timepoint = 3),
      broom::tidy(TTRthreshModelCombined4) %>%
        mutate(timepoint = 4),
      broom::tidy(TTRthreshModelCombined5) %>%
        mutate(timepoint = 5)) %>%
  filter(., term == "as.factor(population)WA") %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("population-comparison-TTRthresh-stats.csv", quote = FALSE, row.names = FALSE) #Combine tidy output from GLMs. Filter out population comparison statistical information and correct for multiple comparisons. Save output as a .csv
```
