---
title: "03-TTR-analysis-WA"
author: "Yaamini Venkataraman"
date: '2025-04-01'
output: html_document
---

In this document I'll examine how time-to-right (TTR) varies by temperature and time for WA, ME, and NH crabs. Parasite surveys were also conducted for these crabs.

# Set up R Markdown document

```{bash}
mkdir ../../output/mult-pop-heat
mkdir ../../output/mult-pop-heat/03-TTR-analysis
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../../output/mult-pop-heat/03-TTR-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, include=FALSE}
#install.packages("tidyverse")
#install.packages("plotrix")
#install.packages("RColorBrewer")
#install.packages("lme4")
#install.packages("ggpubr")
#install.packages("patchwork")
#install.packages("respirometry")
require(tidyverse)
require(plotrix)
require(RColorBrewer)
require(lme4)
require(ggpubr)
require(patchwork)
require(respirometry)
```

```{r}
sessionInfo()
```

# Import data

## WA

```{r}
rawTTRWA <- read.csv("../../../data/mult-pop-heat/WA_HSP_TTR.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRWA)
```

```{r}
genotypeWA <- read.csv("../../../data/mult-pop-heat/WA_genotype.csv", header = TRUE) %>%
  dplyr::select(., -c(extraction.date:gel.date, notes)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeWA) #Confirm import
```

## ME

```{r}
rawTTRME <- read.csv("../../../data/mult-pop-heat/ME_HSP_TTR.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRME)
```

```{r}
genotypeME <- read.csv("../../../data/mult-pop-heat/ME_genotype.csv", header = TRUE) %>%
  dplyr::select(., -c(extraction.date:gel.date, notes)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is numeric. Add columns for precense of C and T
head(genotypeME) #Confirm import
```

```{r}
parasiteME <- read.csv("../../../data/mult-pop-heat/ME_HSP_Para.csv", header = TRUE) %>%
  mutate(., Intensity = Meta/CW) %>%
  dplyr::select(., -c(Species:CW)) #Import parasite data and calculate intensity. Select columns of interest.
head(parasiteME) #Confirm import
```

## NH

```{r}
rawTTRNH <- read.csv("../../../data/mult-pop-heat/NH_HSP_TTR.csv", header = TRUE, na.strings = c("N/A", "NA")) #Import raw data
head(rawTTRNH)
```

```{r}
genotypeNH <- read.csv("../../../data/mult-pop-heat/NH_genotype.csv", header = TRUE) %>%
  dplyr::select(., -c(extraction.date:gel.date, notes)) %>%
  mutate(., presenceC = case_when(genotype == "CC" ~ "Y",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "N")) %>%
  mutate(., presenceT = case_when(genotype == "CC" ~ "N",
                                  genotype == "CT" ~ "Y",
                                  genotype == "TT" ~ "Y")) #Import genotype data. Remove unnecessary columns. Create a crab.ID column to match with other data and ensure it is nuNHric. Add columns for precense of C and T
head(genotypeNH) #Confirm import
```

```{r}
parasiteNH <- read.csv("../../../data/mult-pop-heat/NH_HSP_Para.csv", header = TRUE) %>%
  dplyr::select(., -c(Species:CW)) #Import parasite data and select columns of interest
head(parasiteNH) #Confirm import
```

# Format data

```{r}
modTTRWA <- rawTTRWA %>%
  dplyr::select(., -notes) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeWA %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTRWA) #Confirm formatting
```

```{r}
modTTRME <- rawTTRME %>%
  dplyr::select(., -notes) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeME %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  left_join(x = ., y = parasiteME, by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTRME) #Confirm formatting
```

```{r}
modTTRNH <- rawTTRNH %>%
  dplyr::select(., -notes) %>%
  mutate(., integument.cont = case_when(integument.color == "BG" ~ 0.5,
                                        integument.color == "G" ~ 1.0,
                                        integument.color == "YG" ~ 1.5,
                                        integument.color == "Y" ~ 2.0,
                                        integument.color == "YO" ~ 2.5,
                                        integument.color == "O" ~ 3,
                                        integument.color == "RO" ~ 3.5,
                                        integument.color == "R" ~ 4)) %>%
  mutate(., trial.1 = na_if(trial.1, 91.00)) %>%
  mutate(., trial.2 = na_if(trial.2, 91.00)) %>%
  mutate(., trial.3 = na_if(trial.3, 91.00)) %>%
  rowwise(.) %>% 
  mutate(., TTRavg = mean(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  mutate(., TTRSE = std.error(c(trial.1, trial.2, trial.3), na.rm = TRUE)) %>%
  left_join(x = ., y = (genotypeNH %>% dplyr::select(crab.ID, genotype:presenceT)), by = "crab.ID") %>%
  left_join(x = ., y = parasiteNH, by = "crab.ID") %>%
  ungroup(.) #Remove sampling ID and notes columns. Use case_when to create a column of continuous integument color metrics. Change all 91 to NA, calculate average and SE using data from the three TTR trials using rowwise operations. Ungroup. Join with genotype data.
head(modTTRNH) #Confirm formatting
```

# Data exploration

```{bash}
mkdir figures
```

```{r}
plotColors <- c(brewer.pal(9, "Greens")[7],
                brewer.pal(9, "Oranges")[7],
                brewer.pal(9, "Purples")[7]) #Create color scheme
```

## Sex

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = sex)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-sex-WA.pdf", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = sex)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-sex-ME.pdf", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = sex)) +
  geom_jitter(alpha = 0.75) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-sex-NH.pdf", height = 8.5, width = 11)
```

## Integument color

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = integument.color)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  scale_color_manual(values = c("#66C2A5","#A6D96A","#D9EF8B","#FFFF99","#FDAE61","#F46D43","#D53E4F"),
                     name = "Integument",
                     breaks = c("BG", "G", "YG", "Y", "YO", "O", "RO"),
                     labels = c("BG", "G", "YG", "Y", "YO", "O", "RO")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-integument-WA.pdf", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = integument.color)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  scale_color_manual(values = c("#66C2A5","#A6D96A","#D9EF8B","#FFFF99","#FDAE61","#F46D43","#D53E4F"),
                     name = "Integument",
                     breaks = c("BG", "G", "YG", "Y", "YO", "O", "RO"),
                     labels = c("BG", "G", "YG", "Y", "YO", "O", "RO")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-integument-ME.pdf", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = integument.color)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  scale_color_manual(values = c("#66C2A5","#A6D96A","#D9EF8B","#FFFF99","#FDAE61","#F46D43","#D53E4F"),
                     name = "Integument",
                     breaks = c("BG", "G", "YG", "Y", "YO", "O", "RO"),
                     labels = c("BG", "G", "YG", "Y", "YO", "O", "RO")) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-integument-NH.pdf", height = 8.5, width = 11)
```

## Missing swimmer

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = as.factor(missing.swimmer))) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-missswim-WA.pdf", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = as.factor(missing.swimmer))) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-missswim-ME.pdf", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = as.factor(missing.swimmer))) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-missswim-NH.pdf", height = 8.5, width = 11)
```

## Weight

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = weight)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-weight-WA.pdf", height = 8.5, width = 11)
```

```{r}
modTTRWA %>%
  filter(timepoint == 0) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight
#Weight = 25.11071 g ± 11.35336 g
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = weight)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-weight-ME.pdf", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  filter(timepoint == 0) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight
#Weight = 28.094 g ± 7.443818 g
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = weight)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-weight-NH.pdf", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  filter(timepoint == 0) %>%
  summarize(., weight_mean = mean(weight, na.rm = TRUE),
            weight_sd = sd(weight, na.rm = TRUE)) #Calculate mean ± SD for weight
#Weight = 73.25467 g ± 21.15626 g
```

## Carapace width

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = carapace.width)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-carapaceWidth-WA.pdf", height = 8.5, width = 11)
```

```{r}
modTTRWA %>%
  filter(timepoint == 0) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE),
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width
#CW = 48.92857 mm ± 5.623552 mm
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = carapace.width)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-carapaceWidth-ME.pdf", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  filter(timepoint == 0) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE),
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width
#CW = 50.76667 mm ± 4.11627 mm
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = carapace.width)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 100, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-carapaceWidth-NH.pdf", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  filter(timepoint == 0) %>%
  summarize(., cw_mean = mean(carapace.width, na.rm = TRUE),
            cw_sd = sd(carapace.width, na.rm = TRUE)) #Calculate mean ± SD for carapace width
#CW = 68.53333 mm ± 7.443757 mm
```

## Genotype

```{r}
modTTRWA %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = genotype)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRWA$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 95, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-genotype-WA.pdf", height = 8.5, width = 11)
```

```{r}
genotypeWA %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart-WA.jpg", height = 8.5, width = 11)
```

```{r}
modTTRME %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = genotype)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRME$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 95, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-genotype-ME.pdf", height = 8.5, width = 11)
```

```{r}
genotypeME %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart-ME.jpg", height = 8.5, width = 11)
```

```{r}
modTTRNH %>%
  ggplot(., mapping = aes(x = timepoint, y = TTRavg, color = genotype)) +
  geom_jitter(alpha = 0.8) +
  scale_x_continuous(name = "timepoint",
                     breaks = unique(modTTRNH$timepoint)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = seq(0, 95, 10)) +
  theme_classic(base_size = 15)
ggsave("figures/time-to-right-genotype-NH.pdf", height = 8.5, width = 11)
```

```{r}
genotypeNH %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = genotype)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "Genotype",
                    values = c("grey90", "grey45", "grey0")) +
  theme_void(base_size = 15) #Group data by genotype and summarize. Plot counts as a pie chart.
ggsave("figures/genotype-pie-chart-NH.jpg", height = 8.5, width = 11)
```

### Hardy-Weinberg equilibrium

Since I have the genotype data, I want to see if my population is in Hardy-Weinbergy equilibrium. I used [this calculator](https://www.cog-genomics.org/software/stats) to obtain p-values.

```{r}
genSummarizedWA <- genotypeWA %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n())
genSummarizedWA
```

WA population p-value: 0.21618933186133

```{r}
genSummarizedME <- genotypeME %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n())
genSummarizedME
```

Full population p-value: 0.64694443906897

```{r}
genSummarizedNH <- genotypeNH %>%
  filter(., is.na(genotype) == FALSE) %>%
  group_by(., genotype) %>%
  summarize(., count = n())
genSummarizedNH
```

Full population p-value: 1

All three populations used in this experiment are in Hardy-Weinberg equilibrium. 

# Assess differences in TTR based on time

I will use a linear mixed effects model with individual as a random effect to analyze the data, following [this tutorial](https://jontalle.web.engr.illinois.edu/MISC/lme4/bw_LME_tutorial.pdf).

## WA

```{r}
hist(modTTRWA$TTRavg)
hist(x = log(modTTRWA$TTRavg)) #Log transformation helps normalize data
```

```{r}
TTRmodelWA <- lmer(log(TTRavg) ~ timepoint + 
                     as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = modTTRWA, REML = FALSE) #Run model with timepoint, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later.
summary(TTRmodelWA) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
var.lmerWA <- c(0.001317, 0.240556) #Save variances of random effects as a new vector (crab ID, residuals)
percentvar.lmerWA <- (100*var.lmerWA)/sum(var.lmerWA) #Calculate percent variances
percentvar.lmerWA[1] #Crab ID accounts for 0.5445006% of variance
```

Now I want to test the importance of timepoint. Timepoint is ~22 hours of heat exposure. I'll include all other factors in the null model to account for any variation they may contribute.

```{r}
TTRnullTimepointWA <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = modTTRWA, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullTimepointWA, TTRmodelWA) #Timepoint is a significant predictor
```

## ME

```{r}
hist(modTTRME$TTRavg)
hist(x = log(modTTRME$TTRavg)) #Log transformation helps normalize data
```

```{r}
modTTRME
```

```{r}
TTRmodelME <- lmer(log(TTRavg) ~ timepoint + 
                     as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     Intensity +
                     (1|crab.ID), 
                   data = modTTRME, REML = FALSE) #Run model with timepoint, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Also include parasite intensity. Change the likelihood estimator (REML = FALSE) to compare models with LRT later.
summary(TTRmodelME) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
var.lmerME <- c(0.06281, 0.16365) #Save variances of random effects as a new vector (crab ID, residuals)
percentvar.lmerME <- (100*var.lmerME)/sum(var.lmerME) #Calculate percent variances
percentvar.lmerME[1] #Crab ID accounts for 27.73558% of variance
```

```{r}
TTRnullTimepointME <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + Intensity + (1|crab.ID), data = modTTRME, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullTimepointME, TTRmodelME) #Timepoint is a significant predictor
```

```{r}
TTRnullIntensityME <- lmer(log(TTRavg) ~ timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = modTTRME, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullIntensityME, TTRmodelME) #Parasite intensity is not a significant predictor
```

## NH

```{r}
hist(modTTRNH$TTRavg)
hist(x = log(modTTRNH$TTRavg)) #Log transformation helps normalize data
```

```{r}
modTTRNH <- modTTRNH %>%
  filter(., crab.ID != "NH-76") #Remove NH-76, since it died during heat exposure.

modTTRNH
```

ADD INTENSITY TO MODEL WHEN I HAVE THE CORRECT PARASITE INFORMATION.

```{r}
TTRmodelNH <- lmer(log(TTRavg) ~ timepoint + 
                     as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                     (1|crab.ID), 
                   data = modTTRNH, REML = FALSE) #Run model with timepoint, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later.
summary(TTRmodelNH) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
var.lmerNH <- c(0.07948, 0.93137) #Save variances of random effects as a new vector (crab ID, residuals)
percentvar.lmerNH <- (100*var.lmerNH)/sum(var.lmerNH) #Calculate percent variances
percentvar.lmerNH[1] #Crab ID accounts for 7.86269% of variance
```

ADD INTENSITY WHEN I HAVE THE CORRECT PARASITE INFORMATION.

```{r}
TTRnullTimepointNH <- lmer(log(TTRavg) ~ as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = modTTRNH, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullTimepointNH, TTRmodelNH) #Timepoint is a not a significant predictor
```

```{r}
TTRnullIntensityME <- lmer(log(TTRavg) ~ timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + (1|crab.ID), data = modTTRME, REML = FALSE) #Run null model without timepoint to identify significance of predictor
anova(TTRnullIntensityME, TTRmodelME) #Parasite intensity is not a significant predictor
```

```{r}
TTRModelComparisons <- rbind(broom::tidy(anova(TTRnullTimepointWA, TTRmodelWA)),
                             broom::tidy(anova(TTRnullTimepointME, TTRmodelME)),
                             broom::tidy(anova(TTRnullIntensityME, TTRmodelME)),                           
                             # broom::tidy(anova(TTRnullIntensityNH, TTRmodelME)),
                             broom::tidy(anova(TTRnullTimepointNH, TTRmodelNH))) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) #Create table for model comparison output. Adjust for multiple comparisons.
TTRModelComparisons #Confirm modifications
```

```{r}
write.csv(TTRModelComparisons, "ttr-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save statistical output
```

### Calculate average values

```{r}
rbind(modTTRWA %>%
        group_by(., timepoint) %>%
        summarize(., mean = mean(TTRavg, na.rm = TRUE), sd = sd(TTRavg, na.rm = TRUE)) %>%
        mutate(., population = "WA"),
      modTTRME %>%
        group_by(., timepoint) %>%
        summarize(., mean = mean(TTRavg, na.rm = TRUE), sd = sd(TTRavg, na.rm = TRUE)) %>%
        mutate(., population = "ME"),
      modTTRNH %>%
        group_by(., timepoint) %>%
        summarize(., mean = mean(TTRavg, na.rm = TRUE), sd = sd(TTRavg, na.rm = TRUE)) %>%
        mutate(., population = "NH")) %>%
  write.csv("average-TTR-calculations.csv", quote = FALSE, row.names = FALSE) #Calculate mean and SD for TTR at each time point for populations. Save as a csv
```

# CODE IN PROGRESS












# Effect of genotype

## Impact on average TTR

Now that I know treatment, time, and their interaction significantly impact TTR, I want to add in genotype information.

```{r}
TTRmodelReduced <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with treatment, timepoint, their interaction, sex, integument color, carapace width, weight, and missing swimmer information as fixed effects, and individual as a random effect. Change the likelihood estimator (REML = FALSE) to compare models with LRT later. Use data without rows with NAs
summary(TTRmodelReduced) #Look at model output.
#Residuals explain more random effect variation than crab ID, crab ID explains very little variation
```

```{r}
TTRfullGenotype <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(genotype) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRfullGenotype) #No significant impact of genotype
```

```{r}
TTRpresenceC <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceC) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRpresenceC) #No significant impact of C allele
```

```{r}
TTRpresenceT <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + as.factor(presenceT) + (1|crab.ID), data = (modTTR %>% filter(., is.na(TTRavg) == FALSE) %>% filter(., is.na(genotype) == FALSE)), REML = FALSE) #Run model with genotype to identify significance of predictor
anova(TTRmodelReduced, TTRpresenceT) #No significant impact of T allele
```

```{r}
TTRModelComparisons<- rbind(broom::tidy(anova(TTRnullTreatment, TTRmodel)),
                            broom::tidy(anova(TTRnullTimepoint, TTRmodel)),
                            broom::tidy(anova(TTRnullInteraction, TTRmodel)),
                            broom::tidy(anova(TTRmodelReduced, TTRfullGenotype)),
                            broom::tidy(anova(TTRmodelReduced, TTRpresenceC)),
                            broom::tidy(anova(TTRmodelReduced, TTRpresenceT))) #Create table for model comparison output
TTRModelComparisons
```

```{r}
TTRModelComparisons <- TTRModelComparisons %>%
  mutate(p.adj = p.adjust(p.value, method = "bonferroni")) #Adjust p-values
```

```{r}
write.csv(TTRModelComparisons, "ttr-model-comparison-stat-output.csv", quote = FALSE, row.names = FALSE) #Save model output
```

There is no significant impact of genotype on average TTR. There are some additional factors I want to understand: how genotype impacts whether or not a crab flipped, and how genotype impacted a difference in TTR.

# Plots

## Average TTR plot

```{r}
modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  theme_classic(base_size = 15) #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
ggsave("figures/time-to-right-avg-boxplot.pdf", height = 8.5, width = 11)
```

## Multipanel plot

```{r}
averageTTRWA <- modTTR %>%
  dplyr::select(., c(timepoint, treatment, TTRavg)) %>%
  filter(., is.na(TTRavg) == FALSE) %>%
  distinct(.) %>%
  ggplot(mapping = aes(x = as.character(timepoint), y = TTRavg, color = treatment, shape = treatment)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_jitter(position = position_dodge(width = 0.75)) +
  scale_y_continuous(name = "Average Time-to-Right (s)",
                     breaks = c(seq(0, 20, 5), seq(20, 100, 20))) +
  scale_x_discrete(name = "Exposure Time (Hours)",
                   breaks = unique(modTTR$timepoint),
                   labels = c(0, 4, 22, 28, 46, 94)) +
  scale_color_manual(values = c(plotColors[1], plotColors[2]),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  scale_shape_manual(values = c(15, 19),
                     name = "Temperature (ºC)",
                     breaks = c("1.5C", "5C"),
                     labels = c("2.1", "5.2")) +
  ggtitle("A. Average Righting Response for WA Crabs") +
  theme_classic(base_size = 15) + theme(legend.position = "bottom") #Select unique timepoint, treatment, and TTRavg data, and remove rows where TTRavg = NA. Plot average TTR for each crab in a boxplot. Do not show outliers with the OG boxplot, but add them in with geom_jitter. Assign colors and shapes to each treatment. Increase base font size.
```

```{r}
failureToRightWA <- modTTR %>%
  group_by(., timepoint, treatment) %>%
  summarize(., total = n(), 
            No = sum(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE),
            Yes = sum(!(is.na(trial.1) == TRUE | is.na(trial.2) == TRUE | is.na(trial.3) == TRUE))) %>%
  pivot_longer(., cols = c(Yes, No), 
               values_to = "count", names_to = "condition") %>%
  mutate(., condition = case_when(condition == "No" ~ "2N",
                                  condition == "Yes" ~ "1Y")) %>%
  ggplot(., aes(x = timepoint, y = count, fill = condition)) +
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~treatment, labeller = labeller(treatment = treatment_labels)) +
  scale_x_continuous(name = "Exposure Time (Hours)",
                     breaks = c(0:5),
                     labels = c(0, 4, 22, 28, 46, 94)) +
  scale_y_continuous(name = "Proportion of Crabs") +
  scale_fill_manual(name = "",
                    values = c("grey70", plotColors[1]),
                    labels = c("Yes", "No")) +
  ggtitle("B. Proportion of WA Crabs that Righted") +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(colour = "white"),
                                        legend.position = "bottom")
```

```{r}
averageTTRWA / failureToRightWA
ggsave("figures/multipanel-WA-plot.pdf", height = 11, width = 8.5)
```


# Population comparison

## Format data

```{r}
modTTRCombined <- rbind(modTTRMA, modTTRWA) %>%
  dplyr::select(., treatment, timepoint, crab.ID, 
                TTRavg, TTRthresh, 
                population, 
                sex, integument.cont, carapace.width, weight, missing.swimmer, genotype, presenceC, presenceT) #Rbind data from both populations, and select columns of interest to use in the model
```

## Run model

```{r}
TTRCombinedNull <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                          as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                          (1|crab.ID), 
                        data = (modTTRCombined %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Null model without population
summary(TTRCombinedNull) #Crab ID doesn't explain any variation!
```

```{r}
TTRPopulation <- lmer(log(TTRavg) ~ treatment + timepoint + treatment:timepoint + 
                        as.factor(population) +
                        as.factor(sex) + integument.cont + carapace.width + weight + as.factor(missing.swimmer) + 
                        (1|crab.ID), 
                      data = (modTTRCombined %>% filter(., is.na(TTRavg) == FALSE)), REML = FALSE) #Model with population
anova(TTRCombinedNull, TTRPopulation) #Population is not a significant predictor of avg TTR responses!
```

Population is not a predictor of average TTR responses! 

